<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Simon Thornley and John Sluyter" />


<meta name="progressive" content="true" />
<meta name="allow-skip" content="true" />
<meta name="learnr-version-prerender" content="0.11.5.9000" />

<title>POPLHLTH 304 tutorial 6: count regression</title>

<!-- header-includes START -->
<!-- HEAD_CONTENT -->
<!-- header-includes END -->
<!-- HEAD_CONTENT -->

<!-- highlightjs -->
<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>


<!-- taken from https://github.com/rstudio/rmarkdown/blob/de8a9c38618903627ca509f5401d50a0876079f7/inst/rmd/h/default.html#L293-L343 -->
<!-- tabsets -->
<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>
<!-- end tabsets -->


</head>

<body>
<a class='sr-only sr-only-focusable visually-hidden-focusable' href='#learnr-tutorial-content'>Skip to Tutorial Content</a>



<div class="pageContent band">
<main class="bandContent page">

<article class="topics" id="learnr-tutorial-content">

<center>
<p>Scabies swimming in mineral oil at 40x magnification.</p>
<div class="vembedr">
<div>
<iframe src="https://www.youtube.com/embed/tR0p9s2Lzuk" width="533" height="300" frameborder="0" allowfullscreen="" data-external="1"></iframe>
</div>
</div>
<p>Live mites on human skin (courtesy of Berghofer QIMR).</p>
<div class="vembedr">
<div>
<iframe src="https://www.youtube.com/embed/aarPJyjkW2w" width="533" height="300" frameborder="0" allowfullscreen="" data-external="1"></iframe>
</div>
</div>
<!-- [![Live mites](http://img.youtube.com/vi/aarPJyjkW2w/0.jpg)](https://youtu.be/aarPJyjkW2w){target="_blank" width="600"} -->
</center>
<div id="section-getting-started" class="section level2">
<h2>Getting started</h2>
<p>This session, we will focus on modelling of count outcomes in
epidemiology.</p>
<p>We will learn how to:</p>
<ul>
<li>Plot the relationships between count variables and other
variables</li>
<li>Run and interpret regression models</li>
<li>Interpret the model fit and summary statistics of models</li>
<li>Check models for over-dispersion</li>
<li>Consider the effects of multi-collinearity on exposure-disease
relationships</li>
<li>Use regression nomograms to better understand and communicate the
results of regression models <br></br></li>
</ul>
<p>We will be leaning on the following libraries to do these
analyses:</p>
<ul>
<li><a href="https://cran.r-project.org/web/packages/regplot/index.html"
target="_blank"><code>regplot</code></a></li>
<li><a
href="https://cran.r-project.org/web/packages/PerformanceAnalytics/index.html"
target="_blank"><code>PerformanceAnalytics</code></a></li>
<li><a href="https://jtools.jacob-long.com/"
target="_blank"><code>jtools</code></a></li>
<li><a href="https://r-coder.com/correlation-plot-r/"
target="_blank"><code>corrplot</code></a></li>
<li><a href="https://finalfit.org/"
target="_blank"><code>finalfit</code></a> and</li>
<li><a
href="https://socialsciences.mcmaster.ca/jfox/Books/Companion/index.html"
target="_blank"><code>car</code></a></li>
</ul>
</div>
<div id="section-background" class="section level2">
<h2>Background</h2>
<p>We will be diving into a real-life research battle ground today.</p>
<center>
<div class="float">
<img src="images/sander-meyer-kcwkT7mc0fU-unsplash.jpg" title="battle"
width="200" alt="Research battle by Sander Meyer from unsplash.com" />
<div class="figcaption">Research battle by Sander Meyer from
unsplash.com</div>
</div>
</center>
<p>There is some controversy about the cause of acute rheumatic fever,
which is overall rare disease, but it occurs much more commonly in Māori
and Pacific children than children of any other ethnic group in NZ.</p>
<p>This disease often presents with fever and inflamed joints and can
lead to life-limiting sequelae which include thickened and dysfunctional
heart valves. If these are bad enough, it may lead to the need for
cardiac surgery and in some cases, premature death.</p>
<center>
<div class="float">
<img src="images/arf.jfif" width="300"
alt="Thickened heart valves associated with rheumatic heart disease" />
<div class="figcaption">Thickened heart valves associated with rheumatic
heart disease</div>
</div>
</center>
<p>The most prevalent explanation in the health sector is that it is
caused by bacterial streptococcal throat infection.</p>
<p>An alternative explanation is that it is caused by the common skin
condition, <em>scabies</em>, which has the following skin appearance,
which is easy not to notice. The little crops of raised lumps are
scabies and they are caused by the microscopic mite pictured above.</p>
<center>
<div class="float">
<img src="images/Picture2.jpg" width="300"
alt="Typical skin appearance of scabies lesions: image courtesy of Dr Daniel Engelman" />
<div class="figcaption">Typical skin appearance of scabies lesions:
image courtesy of Dr Daniel Engelman</div>
</div>
</center>
<p>The treatment for scabies in New Zealand is pictured below, and we
will use records of pharmacy dispensing of this drug as an indirect
measure of scabies prevalence in the community.</p>
<center>
<div class="float">
<img src="images/permethrin.jfif" width="300" alt="Permethrin lotion" />
<div class="figcaption">Permethrin lotion</div>
</div>
</center>
<p>The proposed mechanism by which scabies causes streptococcal
infection and rheumatic fever is given below:</p>
<p><a
href="https://linkinghub.elsevier.com/retrieve/pii/S0140-6736(19)31136-5"><img
src="images/Rheumatic_fever_theory.png" width="700"
alt="Consequences of scabies infection. Lancet. 2019 Jul 6;394(10192):81-92. doi: 10.1016/S0140-6736(19)31136-5." /></a></p>
<p><a
href="https://linkinghub.elsevier.com/retrieve/pii/S0140-6736(19)31136-5"
target="_blank">Consequences of scabies infection</a></p>
<p>We will be look for the evidence that prescribing for a certain
treatment for scabies (the lotion <strong>permethrin</strong>) is linked
with <strong>acute rheumatic fever</strong> at a <strong>suburb
level</strong> (1 row of data for each suburb). Since the counts of
acute rheumatic fever over a certain period constitute a ‘count’
variable, <strong>Poisson</strong> regression will be used. The
population of the suburb will be used as an <strong>offset</strong> in
the model. An <a
href="https://stats.stackexchange.com/questions/175349/in-a-poisson-model-what-is-the-difference-between-using-time-as-a-covariate-or"
target="_blank"><strong>offset</strong></a> is often used as the
denominator when rate values are modelled. For example, here, our
outcome is the <strong>count or number of permethrin scripts dispensed
in a suburb</strong> divided by the <strong>population of that
suburb</strong>. The count of prescriptions is naturally influenced by
the population, so this information is incorporated in the model as an
offset. The offset allows us to model rates rather than counts.
Technically, it is a covariate or independent variable whose value of
beta coefficient is forced to be one.</p>
<p>Covariates, or potential confounders in the analysis include:</p>
<ul>
<li>The proportion of Māori and Pacific people living in a suburb</li>
<li>Socioeconomic deprivation</li>
</ul>
<p>We will use the counts of acute rheumatic fever by suburb (or census
area unit). This information is derived from routine datasets used in
the health sector. The formal write-up of the analysis is <a
href="https://onlinelibrary.wiley.com/doi/epdf/10.1111/jpc.14697"
target="_blank">here</a>.</p>
<div id="section-brief-revision" class="section level4">
<h4>Brief revision</h4>
<p>Count regression is used when an outcome is an integer or count,
rather than a continuous measure.</p>
<p>Examples of counts include:</p>
<ul>
<li>the number of hospital admissions per day</li>
<li>the number of people who die each week</li>
<li>the number of teeth in a person’s mouth that affected by decay</li>
</ul>
<p>In comparison, continuous measures can include fractions or decimals,
such as:</p>
<ul>
<li>height</li>
<li>weight</li>
<li>blood pressure</li>
<li>age.</li>
</ul>
<p>Count variables have special characteristics of their distribution
which makes linear regression unsuitable. First of all, the distribution
is positive, asymmetric and right skewed, meaning a log transformation
of the count is necessary. Also, the variance is usually proportional to
the count, rather than a constant.</p>
</div>
<div id="section-lecture-revision" class="section level3">
<h3>Lecture revision</h3>
<p>Our outcome is acute rheumatic fever counts and we will build models
for this using regression. For the following 3 questions, please answer
<em>True</em> of <em>False</em> in response to each statement.</p>
<div id="section-question-1" class="section level5">
<h5><strong>Question 1</strong></h5>
<div class="panel panel-default tutorial-question-container">
<div data-label="revision_1" class="tutorial-question panel-body">
<div id="revision_1-answer_container" class="shiny-html-output"></div>
<div id="revision_1-message_container" class="shiny-html-output"></div>
<div id="revision_1-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
</div>
<div id="section-question-2" class="section level5">
<h5><strong>Question 2</strong></h5>
<div class="panel panel-default tutorial-question-container">
<div data-label="revision_2" class="tutorial-question panel-body">
<div id="revision_2-answer_container" class="shiny-html-output"></div>
<div id="revision_2-message_container" class="shiny-html-output"></div>
<div id="revision_2-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
</div>
<div id="section-question-3" class="section level5">
<h5><strong>Question 3</strong></h5>
<div class="panel panel-default tutorial-question-container">
<div data-label="revision_3" class="tutorial-question panel-body">
<div id="revision_3-answer_container" class="shiny-html-output"></div>
<div id="revision_3-message_container" class="shiny-html-output"></div>
<div id="revision_3-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
</div>
</div>
</div>
<div id="section-set-up" class="section level2">
<h2><strong>Set-up</strong></h2>
<p>To set-up our session for analysis and call the necessary libraries,
we’ll use <code>pacman::p_load()</code> instead of
<code>install.packages()</code> and <code>library()</code>. The
<code>pacman</code> library shortens this procedure, and
<code>p_load()</code> will look for a package and install it if it can’t
be found. It will also put it on the search path, like the
<code>library()</code> command does. We won’t run this code, as it will
mess up our tutorial, but it is essential for getting this running on a
local machine or posit cloud.</p>
<pre class="r"><code>if(!require(pacman)) install.packages(&quot;pacman&quot;)
pacman::p_load(tidyverse, 
               epiDisplay,
               visdat,
               skimr,
               lattice,
               jtools,
               ggstance,
               broom.mixed,
               finalfit,
               dplyr,
               kableExtra,
               car,
               MixAll,
               regplot,
               PerformanceAnalytics,
               corrplot)</code></pre>
<p>View the dataframe; called <code>for_shiny</code>. Notice that
there’s 1 row for each suburb:</p>
<div class="tutorial-exercise" data-label="view_1" data-completion="1"
data-diagnostics="1" data-startover="1" data-lines="0"
data-pipe="|&gt;">
<pre class="text"><code>for_shiny</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
List the variable names:
<div class="tutorial-exercise" data-label="names_1" data-completion="1"
data-diagnostics="1" data-startover="1" data-lines="0"
data-pipe="|&gt;">
<pre class="text"><code>for_shiny |&gt; names()</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
Rename our dataframe as <code>df</code>. Then rename the variables so
they’re more understandable:
<div class="tutorial-exercise" data-label="rename_1" data-completion="1"
data-diagnostics="1" data-startover="1" data-lines="0"
data-pipe="|&gt;">
<pre class="text"><code>df &lt;- for_shiny # Rename dataframe

# Rename variables
names(df) &lt;- c(&quot;arf_count&quot;, &quot;population&quot;, &quot;Pacific_proportion&quot;,
               &quot;Maori_proportion&quot;, &quot;Permethrin_rate&quot;, &quot;Suburb&quot;,
               &quot;nzdep_decile&quot;, &quot;arf_rate&quot;)
df |&gt; names() # Check that the variables have been renamed</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div id="section-data-integrity-checks" class="section level3">
<h3><strong>Data integrity checks</strong></h3>
</div>
<div id="section-duplicates" class="section level3">
<h3>1) Duplicates</h3>
There should be 1 row for each <code>Suburb</code>, so check if this
variable is duplicated.
<div class="tutorial-exercise" data-label="duplicate_1"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>stopifnot(df[df$Suburb |&gt; duplicated(),] == 0)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<p>If <code>Suburb</code> is duplicated, the R output would read:
<code>df[df$Suburb |&gt; duplicated(),] == 0 are not all TRUE</code>.<br />
Are there any duplicates?</p>
</div>
<div id="section-ranges" class="section level3">
<h3>2) Ranges</h3>
<div class="tutorial-exercise" data-label="range_1" data-completion="1"
data-diagnostics="1" data-startover="1" data-lines="0"
data-pipe="|&gt;">
<pre class="text"><code>df |&gt; skim()</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<p>Are there any implausible extreme values (check <code>p0</code>
[minimum] and <code>p100</code> [maximum])?</p>
<div id="section-question-4" class="section level5">
<h5><strong>Question 4</strong></h5>
<p>From the summary data above, what is the <em>standard error</em> of
<code>arf_count</code>?</p>
<p>Hint, you will need to use the square root <code>sqrt()</code>
function and use R as a calculator.</p>
<div class="tutorial-exercise" data-label="variance_1"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":true,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise-support" data-label="variance_1-solution"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>1.17/sqrt(413)</code></pre>
</div>
<div id="section-variance_1-hint">
<p><strong>Hint:</strong> standard error = standard deviation/square
root(number of observations)</p>
</div>
</div>
<div id="section-question-5" class="section level5">
<h5><strong>Question 5</strong></h5>
From the summary data above, what is the <em>variance</em> of
<code>arf_count</code>?
<div class="tutorial-exercise" data-label="variance_2"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":true,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise-support" data-label="variance_2-solution"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>(1.17/sqrt(413))^2</code></pre>
</div>
<div id="section-variance_2-hint">
<p><strong>Hint:</strong> Variance = (standard error)^2</p>
</div>
</div>
</div>
<div id="section-missing-data" class="section level3">
<h3>3) Missing data</h3>
<div class="tutorial-exercise" data-label="missing_1"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>df |&gt; vis_miss(sort_miss = TRUE)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<p>Do you see any missing data?</p>
</div>
</div>
<div id="section-univariable-distributions-histograms"
class="section level2">
<h2><strong>Univariable distributions: histograms</strong></h2>
<p>Use the <code>histogram()</code> function for these.</p>
Outcome as a <em>count</em>: <code>arf_count</code>
<div class="tutorial-exercise" data-label="histogram_1"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>df$arf_count |&gt; histogram()</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
Outcome as a <em>rate</em>: <code>arf_rate</code>
<div class="tutorial-exercise" data-label="histogram_2"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>df$arf_rate |&gt; histogram()</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div id="section-question-6" class="section level5">
<h5><strong>Question 6</strong></h5>
<div class="panel panel-default tutorial-question-container">
<div data-label="histogram_3" class="tutorial-question panel-body">
<div id="histogram_3-answer_container" class="shiny-html-output"></div>
<div id="histogram_3-message_container" class="shiny-html-output"></div>
<div id="histogram_3-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
</div>
<div id="section-question-7" class="section level5">
<h5><strong>Question 7</strong></h5>
<div class="panel panel-default tutorial-question-container">
<div data-label="histogram_4" class="tutorial-question panel-body">
<div id="histogram_4-answer_container" class="shiny-html-output"></div>
<div id="histogram_4-message_container" class="shiny-html-output"></div>
<div id="histogram_4-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
<p>First, we will check the distribution of our outcome</p>
Exposure: <code>permethrin_rate</code>
<div class="tutorial-exercise" data-label="histogram_5"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>df$Permethrin_rate |&gt; histogram()</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
</div>
<div id="section-question-8" class="section level5">
<h5><strong>Question 8</strong></h5>
Then the distribution of our confounders. Adjust the following code to
plot three histograms of the following variables.
<div class="tutorial-exercise" data-label="histogram_6"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>df$Pacific_proportion 
df$Maori_proportion
df$nzdep_decile </code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":true,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise-support" data-label="histogram_6-solution"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>df$Pacific_proportion |&gt; histogram()
df$Maori_proportion |&gt; histogram()
df$nzdep_decile |&gt; histogram()</code></pre>
</div>
<div id="section-histogram_6-hint">
<p><strong>Hint:</strong> You need to use the <code>histogram()</code>
function</p>
</div>
</div>
<div id="section-question-9" class="section level5">
<h5><strong>Question 9</strong></h5>
<div class="panel panel-default tutorial-question-container">
<div data-label="histogram_7" class="tutorial-question panel-body">
<div id="histogram_7-answer_container" class="shiny-html-output"></div>
<div id="histogram_7-message_container" class="shiny-html-output"></div>
<div id="histogram_7-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
When a categorical variable has many groups, it’s common to combine
these to avoid sparse data. For <code>nzdep_decile</code>, which has 10
categories (last histogram above), we’ll combine consecutive pairs to
create quintiles (5 groups). The breaks are included in the category.
So, for example, the first category includes any value from negative
infinity (<code>-Inf</code>) to 2. Any value greater than two will be
incorporated in the next category.
<div class="tutorial-exercise" data-label="histogram_8"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code># 5 groups instead of 10
df$nzDepQuint &lt;- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),
                     labels = c(&quot;1 to 2 (least deprived)&quot;,
                                &quot;3 to 4&quot;,
                                &quot;5 to 6&quot;,
                                &quot;7 to 8&quot;,
                                &quot;9 to 10 (most deprived)&quot;))

# Check the frequency of our 5 groups
epiDisplay::tabpct(df$nzDepQuint, df$nzdep_decile)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
</div>
</div>
<div id="section-bivariate-associations-scatterplots"
class="section level2">
<h2><strong>Bivariate associations: scatterplots</strong></h2>
<p>Before undertaking any regression analysis, it is useful to visualise
the relationships especially between key exposure and outcome
variables.</p>
<p>Use the <code>car::scatterplot()</code> function.</p>
####Exposure and outcome
<div class="tutorial-exercise" data-label="scatterplot_1"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>car::scatterplot(arf_rate ~ Permethrin_rate, data = df)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<p>How would you describe the association between the exposure and
outcome?</p>
<div id="section-question-10" class="section level5">
<h5><strong>Question 10</strong></h5>
<div class="panel panel-default tutorial-question-container">
<div data-label="scatter_1" class="tutorial-question panel-body">
<div id="scatter_1-answer_container" class="shiny-html-output"></div>
<div id="scatter_1-message_container" class="shiny-html-output"></div>
<div id="scatter_1-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
</div>
<div id="section-question-11" class="section level5">
<h5><strong>Question 11</strong></h5>
Complete the following lines of code to make scatter plots between
covariates and the outcome <code>arf_rate</code>.<br />

<div class="tutorial-exercise" data-label="scatterplot_2"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>arf_rate ~ nzdep_decile
arf_rate ~ Pacific_proportion 
arf_rate ~ Maori_proportion </code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":true,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div id="section-scatterplot_2-hint">
<p><strong>Hint:</strong> You need to use the
<code>car::scatterplot()</code> function. You’ll also need to fill the
<code>data = ...</code> argument, so that <code>R</code> knows which
variables in which <code>data.frame</code> you are referring to.</p>
</div>
<div class="tutorial-exercise-support"
data-label="scatterplot_2-solution" data-completion="1"
data-diagnostics="1" data-startover="1" data-lines="0"
data-pipe="|&gt;">
<pre class="text"><code>car::scatterplot(arf_rate ~ nzdep_decile, data = df)
car::scatterplot(arf_rate ~ Pacific_proportion, data = df)
car::scatterplot(arf_rate ~ Maori_proportion, data = df)</code></pre>
</div>
<p>Are the potential confounders associated with the outcome?</p>
</div>
<div id="section-question-12" class="section level5">
<h5><strong>Question 12</strong></h5>
<div class="panel panel-default tutorial-question-container">
<div data-label="scatter_2_int" class="tutorial-question panel-body">
<div id="scatter_2_int-answer_container" class="shiny-html-output"></div>
<div id="scatter_2_int-message_container" class="shiny-html-output"></div>
<div id="scatter_2_int-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
<strong>Quick method to make several scatterplots</strong><br />
Use the <code>PerformanceAnalytics::chart.Correlation()</code> function
to get all correlations in 1 chart. A short-cut! Here, we select five
variables from <code>df</code> and put them into a
<code>data.frame</code> called <code>my_data</code>. The we apply our
<code>PerformanceAnalytics::chart.Correlation()</code> function to
<code>my_data</code> to get histograms, scatter plots and Pearson
correlation coefficients.
<div class="tutorial-exercise" data-label="scatterplot_4"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>## Select variables of interest into new data.frame
my_data &lt;- df[, c(&quot;Pacific_proportion&quot;, &quot;Maori_proportion&quot;, &quot;Permethrin_rate&quot;, &quot;nzdep_decile&quot;, &quot;arf_rate&quot;)]

## Chart the correlations and regressions and distributions of each variable.
PerformanceAnalytics::chart.Correlation(my_data, histogram = TRUE, pch = 19)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
</div>
<div id="section-question-13" class="section level5">
<h5><strong>Question 13</strong></h5>
<div class="panel panel-default tutorial-question-container">
<div data-label="scatterplot_5" class="tutorial-question panel-body">
<div id="scatterplot_5-answer_container" class="shiny-html-output"></div>
<div id="scatterplot_5-message_container" class="shiny-html-output"></div>
<div id="scatterplot_5-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
</div>
</div>
<div id="section-regression" class="section level2">
<h2><strong>Regression</strong></h2>
<p>We will build a Poisson regression model with our outcome as
<code>arf_count</code> and exposure as <code>Permethrin_rate</code>. To
adjust for population size, we will include <code>log_population</code>
as an offset. We will also adjust for NZDep quintile
(<code>nzDepQuint</code>), <code>Pacific_proportion</code> and
<code>Maori_proportion</code>.</p>
<p>Run the following code to make a Poisson regression model of
<code>arf_count</code> with all other risk factors as independent
variables or covariates.</p>
<div class="tutorial-exercise" data-label="regression_1"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>## Make offset variable
df$log_population &lt;- log(df$population)

## Build model
fit &lt;- glm(arf_count ~ Permethrin_rate +
           nzDepQuint +
           Pacific_proportion +
           Maori_proportion, 
           offset = log_population,
           family = poisson(link = &quot;log&quot;),
           data = df)

## Print summary of model
summary(fit)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<p>Look at the term, <code>offset = log_population</code>, in the model.
With counts (<code>arf_count</code>) as the outcome, having
<code>log_population</code> as the offset makes the model represent a
<em>rate</em> (count per unit <em>population</em>).</p>
<p>Check your understanding.</p>
<div id="section-question-14" class="section level5">
<h5><strong>Question 14</strong></h5>
<div class="panel panel-default tutorial-question-container">
<div data-label="regression_2" class="tutorial-question panel-body">
<div id="regression_2-answer_container" class="shiny-html-output"></div>
<div id="regression_2-message_container" class="shiny-html-output"></div>
<div id="regression_2-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
</div>
<div id="section-question-15" class="section level5">
<h5><strong>Question 15</strong></h5>
<div class="panel panel-default tutorial-question-container">
<div data-label="regression_3" class="tutorial-question panel-body">
<div id="regression_3-answer_container" class="shiny-html-output"></div>
<div id="regression_3-message_container" class="shiny-html-output"></div>
<div id="regression_3-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
<p>While the exponentiated coefficient don’t seem large (1.00035 is a
0.035% change, with a one unit increase of permethrin), the nature of
the association may only be understood by also knowing to what extent
permethrin dispensing varies in the population.</p>
<p>From our <code>skim()</code> function used earlier, we can see that
the minimum values of permethrin dispensing is 0 and the maximum is
3,201. Therefore, the ratio of risk for a change of 3,000 is
<code>exp(3.527e-04 * 3000)</code> which equals 2.88. This means that
comparing the lowest to highest suburbs results in a 100 * (2.88 - 1) =
188% change in acute rheumatic fever incidence. That weak association is
now looking <em>very strong</em>.</p>
Exponentiate our model coefficients with the <code>jtools::sum</code>
function:
<div class="tutorial-exercise" data-label="regression_4"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>jtools::summ(fit, exp = TRUE)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
</div>
<div id="section-question-16" class="section level5">
<h5><strong>Question 16</strong></h5>
<div class="panel panel-default tutorial-question-container">
<div data-label="regression_5" class="tutorial-question panel-body">
<div id="regression_5-answer_container" class="shiny-html-output"></div>
<div id="regression_5-message_container" class="shiny-html-output"></div>
<div id="regression_5-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
</div>
<div id="section-question-17" class="section level5">
<h5><strong>Question 17</strong></h5>
We saw before that, for the outcome, there’s evidence of
under-dispersion. To account for this, we will specify a
<em>quasiPoisson</em> model (please use <code>fit_qp</code> as the new
model object. Do this by changing <code>poisson</code> to
<code>quasipoisson</code> in the code below and running it.
<div class="tutorial-exercise" data-label="regression_6"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>fit1 &lt;- glm(arf_count ~ I(Permethrin_rate/1000) +
             nzDepQuint +
             Pacific_proportion +
             Maori_proportion + offset(log(population)),
           family = poisson(link = &quot;log&quot;),
           data = df)
summary(fit1)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":true,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise-support"
data-label="regression_6-solution" data-completion="1"
data-diagnostics="1" data-startover="1" data-lines="0"
data-pipe="|&gt;">
<pre class="text"><code>fit_qp &lt;- glm(arf_count ~ I(Permethrin_rate/1000) +
             nzDepQuint +
             Pacific_proportion +
             Maori_proportion + offset(log(population)),
           family = quasipoisson(link = &quot;log&quot;),
           data = df)
summary(fit_qp)</code></pre>
</div>
<div id="section-regression_6-hint">
<p><strong>Hint:</strong> Please change the <code>family</code>
argument, and change the name of the model object from <code>fit1</code>
to <code>fit_qp</code>.</p>
</div>
</div>
<div id="section-question-18" class="section level5">
<h5><strong>Question 18</strong></h5>
Using the code format below, exponentiate your <span
class="math inline">\(\beta\)</span> coefficients so that you get
incidence rate ratios.
<div class="tutorial-exercise" data-label="regression_7"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code># jtools::summ(model_object, exp = logical_value)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":true,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise-support"
data-label="regression_7-solution" data-completion="1"
data-diagnostics="1" data-startover="1" data-lines="0"
data-pipe="|&gt;">
<pre class="text"><code>jtools::summ(fit_qp, exp = TRUE)</code></pre>
</div>
<div id="section-regression_7-hint">
<p><strong>Hint:</strong> What is the new model object names and
<code>logical_value</code> for this question?</p>
</div>
Now, use the <code>jtools::plot_summs()</code> function to plot the rate
ratios:
<div class="tutorial-exercise" data-label="regression_8"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>jtools::plot_summs(fit_qp, exp = TRUE)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<p>You should get a graph showing rate ratios:</p>
<ul>
<li>Circles for the <code>Estimates</code></li>
<li>Horizontal lines for the 95% confidence intervals</li>
<li>A dotted vertical line for a ratio of 1: the null value</li>
</ul>
<p>Which variables are significantly associated with the outcome?</p>
</div>
</div>
<div id="section-multi-collinearity" class="section level2">
<h2><strong>Multi-collinearity</strong></h2>
<p>In a model, when 2 or more predictors are <em>strongly
correlated</em> with one other, this is called
<strong>multicollinearity</strong>. In a model, it’s a problem as
it:</p>
<ul>
<li>increases the variance of estimates of regression coefficients</li>
<li>makes these estimates unstable and unreliable <br></li>
</ul>
<p>We can detect it by looking for strong correlations between
variables.</p>
<p>Use the <em>cor()</em> function to get the correlation between
<code>Pacific_proportion</code> and <code>Permethrin_rate</code>:</p>
<div class="tutorial-exercise" data-label="multicollinearity_1"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>cor(df$Pacific_proportion, df$Permethrin_rate)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<p>You should get a correlation of 0.88, which is
<strong>strong</strong>. Thus, multicollinearity may be a problem for
our analysis.<br />
To deal with this, we can remove one of these predictors from the model.
So that we retain our exposure (<code>Permethrin_rate</code>) in our
model, let’s remove <code>Pacific_proportion</code> by adding
<code>#</code> before it.<br />
Another change is that we’ll divide <code>Permethrin_rate</code> by
1000. This will increase the coefficient for this variable by 1000 so
that it’s more meaningful. (This is like converting grams into
kilograms; the latter is more meaningful for larger weights). Below, we
use the <code>I()</code> function to do this calculation within the
‘model’ argument.<br />
Now run our revised code below:</p>
<div class="tutorial-exercise" data-label="regression_9"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>df$log_population &lt;- log(df$population)

fit3 &lt;- glm(arf_count ~ I(Permethrin_rate/1000) +
            nzDepQuint +
             #Pacific_proportion +
            Maori_proportion, offset=(log(population)),
           family = quasipoisson(link = &quot;log&quot;),
           data = df)

summary(fit3)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<p>What is the slope for the <code>Permethrin_rate/1000</code> term? Is
it statistically significant?</p>
Run the code for the next model, which includes <code>Permethrin</code>
as a predictor instead of <code>Permethrin_rate/1000.</code>
<div class="tutorial-exercise" data-label="regression_10"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>df$log_population &lt;- log(df$population)

fit4 &lt;- glm(arf_count ~  I(Permethrin_rate/1000) +
              nzDepQuint +
              #Pacific_proportion +
              Maori_proportion, offset=log_population,
              family = quasipoisson(link = &quot;log&quot;),
              data = df)

summary(fit4)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<p>Was is the incidence rate ratio for
<code>I(Permethrin_rate/1000)</code>?</p>
<p>This next model has <code>Pacific_proportion</code> included again.
So we can test for the effect of multi-collinearity. Run the code
below.</p>
<div class="tutorial-exercise" data-label="regression_11"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>fit5 &lt;- glm(arf_count ~ I(Permethrin_rate/1000) +
              nzDepQuint +
              Pacific_proportion +
              Maori_proportion, offset(log(population)),
              family = quasipoisson(link = &quot;log&quot;),
              data = df)

summary(fit5)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<p>Because of multicollinearity, we expect the standard error for
<code>I(Permethrin_rate/1000)</code> to <em>increase</em> after
<code>Pacific_proportion</code> has been added. Has it?</p>
</div>
<div id="section-regression-nomogram" class="section level2">
<h2><strong>Regression nomogram</strong></h2>
This is a visual summary of structure of a model. We will display
regression nomograms by suburb. First, let’s print the names of the
<code>Suburb</code>s in our dataset to see if there is one we recognise.
<div class="tutorial-exercise" data-label="nomogram_1"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>df$Suburb |&gt; print()</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<p>The code below will run in posit, but unfortunately not on this
platform, so we will just show you the code and the output. We have
plotted the values for Otara East, highlighted on the figure in <span
style="color: red;">red</span>.</p>
<pre class="r"><code>regplot::regplot(fit, 
                 observation = df[df$Suburb == &quot;Otara East&quot;,], clickable = FALSE)</code></pre>
<pre><code>## Error in `[.data.frame`(observation, reord): undefined columns selected</code></pre>
<div class="float">
<img src="images/regplot.png" title="Regression illustrated"
alt="Regplot" />
<div class="figcaption">Regplot</div>
</div>
<p>For Otara East, for example, we see that
<code>Pacific_proportion</code> is about 0.8 or 80%. Also, the
dispensing rate of permethrin <code>Permethrin_rate</code> is relatively
high, as expected by the correlation coefficient between this variable
and <code>Pacific_proportion</code> as we discussed earlier. Each
variable contributes a score (<code>β(X-m) terms</code>) and these add
up as a type of weighted average to produce a predicted rate of acute
rheumatic fever (<code>arf_count</code>). Note that for Otara East, the
predicted <code>arf_count</code> is very high.</p>
<p>Please attempt the next 2 questions.</p>
<div id="section-question-19" class="section level5">
<h5><strong>Question 19</strong></h5>
<div class="panel panel-default tutorial-question-container">
<div data-label="nomogram_3" class="tutorial-question panel-body">
<div id="nomogram_3-answer_container" class="shiny-html-output"></div>
<div id="nomogram_3-message_container" class="shiny-html-output"></div>
<div id="nomogram_3-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
<p>(Hint: In the top line, <code>β(X-m) terms</code>, identify the red
dot that’s furthest from 0. Then read vertically down from it)</p>
</div>
<div id="section-question-20" class="section level5">
<h5><strong>Question 20</strong></h5>
<div class="panel panel-default tutorial-question-container">
<div data-label="nomogram_4" class="tutorial-question panel-body">
<div id="nomogram_4-answer_container" class="shiny-html-output"></div>
<div id="nomogram_4-message_container" class="shiny-html-output"></div>
<div id="nomogram_4-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
<p>Hint: In each density plot or histogram, look at where the red dot
(for Otara East) is.</p>
<pre class="r"><code>regplot::regplot(fit, observation = df[df$Suburb == &quot;Remuera South&quot;,])</code></pre>
<pre><code>## Error in `[.data.frame`(observation, reord): undefined columns selected</code></pre>
<div class="float">
<img src="images/remuera_regplot.png" title="Regression illustrated"
alt="Regplot2" />
<div class="figcaption">Regplot2</div>
</div>
</div>
</div>
<div id="section-model-fit" class="section level2">
<h2><strong>Model fit</strong></h2>
<p>How well a model fits the data can be assessed with a
<strong>‘marginal model plot’</strong>, which shows the relationship
between model predicted (model) and observed data (data). In a good
fitting model, we expect to see the two lines closely superimposed on
each other. Poor fit is identified by seeing separation between the two
lines.</p>
Run this code to produce a marginal model plot for our <code>fit</code>
model object:
<div class="tutorial-exercise" data-label="model_fit_1"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>car::marginalModelPlots(fit)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<p>What is the nature of the overall fit of the data to the model?<br />
- <strong>Good fit</strong> is indicated by <strong>good
alignment</strong> between <code>Data</code> (blue line) and the
<code>Model</code> (red line)<br />
- <strong>Poor fit</strong> is indicated by <strong>poor
alignment</strong> between these lines<br />
- Often, for more extreme or outlying values of an exposure, there is
deviation of the <code>Data</code> from the <code>Model</code>. This may
not invalidate your model, but would be important to be aware of, and
may be worthwhile noting this in your results.</p>
<div id="section-question-21" class="section level5">
<h5><strong>Question 21</strong></h5>
<div class="panel panel-default tutorial-question-container">
<div data-label="model_fit_2" class="tutorial-question panel-body">
<div id="model_fit_2-answer_container" class="shiny-html-output"></div>
<div id="model_fit_2-message_container" class="shiny-html-output"></div>
<div id="model_fit_2-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
</div>
</div>
<div id="section-publication-quality-regression-tables"
class="section level2">
<h2>Publication-quality regression tables</h2>
<p>In published articles, it’s common to present univariable
(unadjusted) and multivariable (adjusted) regression coefficients in a
table. To create these, run the following code.</p>
<div class="tutorial-exercise" data-label="publication_1"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0" data-pipe="|&gt;">
<pre class="text"><code>df$Permethrin_rate_per_1000 &lt;- df$Permethrin_rate/1000  # Like converting grams into kilograms
df$log_population &lt;- log(df$population)  # Offset

dependent &lt;- &quot;arf_count&quot;
explanatory &lt;- c(&quot;Pacific_proportion&quot;, &quot;Maori_proportion&quot;, 
                 &quot;Permethrin_rate_per_1000&quot;,&quot;nzdep_decile&quot;)

# `fit_uni` creates univariable (unadjusted) models
fit_uni = df |&gt; 
  glmuni(dependent, explanatory, family = quasipoisson,
         offset = df$log_population) |&gt; 
  fit2df(estimate_name = &quot;Rate ratio (univariable)&quot;)

# `fit_multi` creates a multivariable (adjusted) model
fit_multi = df |&gt; 
  glmmulti(dependent, explanatory, family = quasipoisson, 
           offset = df$log_population) |&gt; 
  fit2df(estimate_name = &quot;Rate ratio (multivariable)&quot;)

df |&gt;
  ## Crosstable
  summary_factorlist(dependent, explanatory, cont = &quot;median&quot;, fit_id=TRUE)  |&gt; 
  
  #### Add univariable table
  ff_merge(fit_uni, estimate_name = &quot;Rate ratio&quot;) |&gt; 
  
  #### Add multivariable table
  ff_merge(fit_multi, estimate_name = &quot;Rate ratio&quot;,
           last_merge = TRUE) |&gt; 
dependent_label(df, dependent) -&gt; regression_table

regress_tab &lt;- knitr::kable(regression_table, row.names = FALSE, 
                            align = c(&quot;l&quot;, &quot;l&quot;, &quot;r&quot;, &quot;r&quot;, &quot;r&quot;))

#### Puts output into viewer so that it can be cut and pasted into Excel, 
#### then Word.
regress_tab |&gt; kableExtra::kable_styling()</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<p>Notice that the adjusted <code>Rate ratio (multivariable)</code> for
<code>Permethrin_rate_per_1000</code> is not statistically significant.
Does that mean permethrin or scabies is not related to acute rheumatic
fever incidence.</p>
<div class="panel panel-default tutorial-question-container">
<div data-label="pub_model" class="tutorial-question panel-body">
<div id="pub_model-answer_container" class="shiny-html-output"></div>
<div id="pub_model-message_container" class="shiny-html-output"></div>
<div id="pub_model-action_button_container" class="shiny-html-output"></div>
<script>if (Tutorial.triggerMathJax) Tutorial.triggerMathJax()</script>
</div>
</div>
</div>
<div id="section-homework" class="section level2">
<h2>Homework</h2>
<p>Select only suburbs that have a socioeconomic deprivation of 6 or
more:<br />
df_dep &lt;- df[df$nzdep_decile &gt;= 6, ]</p>
<p>Then, for this subgroup, repeat the analyses we’ve covered in this
tutorial.</p>
<p>Please have a go at these tasks in posit. 
<script type="application/shiny-prerendered" data-context="server-start">
library(learnr)
library(tufte)
library(rio)
library(visdat)
library(fs)
library(ggplot2)
library(DiagrammeR)
library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/
library(learnr)
library(gradethis)
library(skimr)
library(lattice)
library(epiDisplay)
library(finalfit)
library(vembedr)
library(png)

gradethis::gradethis_setup()
#setwd("./data")
# load("./Cloud/data/ARF_permethrin.rdata")
load("./data/ARF_permethrin.rdata")
# library(readxl)
# df <- read_xlsx("C:\\Users\\jslu002\\Desktop\\for_shiny.xlsx")
df <- data.frame(for_shiny)
df <- df[!duplicated(df),]

names(df) <- c("arf_count", "population", "Pacific_proportion",
               "Maori_proportion", "Permethrin_rate", "Suburb",
               "nzdep_decile", "arf_rate")

## 5 groups instead of 10
df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),
                     labels = c("1 to 2 (least deprived)",
                                "3 to 4",
                                "5 to 6",
                                "7 to 8",
                                "9 to 10 (most deprived)"))

df$log_population <- log(df$population)

fit <- glm(arf_count ~ Permethrin_rate +
             nzDepQuint +
             Pacific_proportion +
             Maori_proportion, offset=log_population,
           family = poisson(link = "log"),
           data = df)

fit1 <- glm(arf_count ~ Permethrin_rate +
             nzDepQuint +
             Pacific_proportion +
             Maori_proportion + offset(log(population)),
           family = quasipoisson(link = "log"),
           data = df)

fit5 <- glm(arf_count ~ Permethrin_rate +
             nzDepQuint +
             Pacific_proportion +
             Maori_proportion + offset(log(population)),
           family = quasipoisson(link = "log"),
           data = df)

mycounter <- 0
Qnum <- function(count = mycounter){
  mycounter <<- mycounter + 1
  paste0("Question ", mycounter)
}

knitr::opts_chunk$set(error = TRUE)
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::register_http_handlers(session, metadata = NULL)
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::prepare_tutorial_state(session)
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::i18n_observe_tutorial_language(input, session)
</script>
 
<script type="application/shiny-prerendered" data-context="server">
session$onSessionEnded(function() {
        learnr:::event_trigger(session, "session_stop")
      })
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_radio", label = "revision_1", question = structure("The measure of association derived from a count model is an odds ratio.", html = TRUE, class = c("html", 
"character")), answers = list(structure(list(id = "lnr_ans_b1a3add", 
    option = "True", value = "True", label = structure("True", html = TRUE, class = c("html", 
    "character")), correct = FALSE, message = structure("No, the odds ratio is the measure of association for logistic regression (next tutorial)", html = TRUE, class = c("html", 
    "character")), type = "literal"), class = c("tutorial_question_answer", 
"tutorial_quiz_answer")), structure(list(id = "lnr_ans_d94383", 
    option = "False", value = "False", label = structure("False", html = TRUE, class = c("html", 
    "character")), correct = TRUE, message = structure("Yes, the odds ratio is the measure of association for logistic regression (next tutorial). For count regression, the measure of association is the <em>incidence rate ratio<\u002fem>", html = TRUE, class = c("html", 
    "character")), type = "literal"), class = c("tutorial_question_answer", 
"tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
"character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
"character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
"character")), try_again = structure("Incorrect", html = TRUE, class = c("html", 
"character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
"character")), message = NULL, post_message = NULL), ids = list(
    answer = "revision_1-answer", question = "revision_1"), loading = NULL, 
    random_answer_order = FALSE, allow_retry = TRUE, seed = 87334706.4593316, 
    options = list()), class = c("learnr_radio", "tutorial_question"
)), session = session)
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_radio", label = "revision_2", question = structure("Linear regression is unsuitable to use for count outcomes as it may predict negative counts", html = TRUE, class = c("html", 
"character")), answers = list(structure(list(id = "lnr_ans_3fee38e", 
    option = "True", value = "True", label = structure("True", html = TRUE, class = c("html", 
    "character")), correct = TRUE, message = structure("Yes, count regression overcomes this problem as the outcome is modelled on a log scale (outcome is log(count)). Exponentiating log(count) predictions gives count predictions that are always positive as exponentiating any non zero number (even negative) gives positive values only", html = TRUE, class = c("html", 
    "character")), type = "literal"), class = c("tutorial_question_answer", 
"tutorial_quiz_answer")), structure(list(id = "lnr_ans_1beaae2", 
    option = "False", value = "False", label = structure("False", html = TRUE, class = c("html", 
    "character")), correct = FALSE, message = structure("No, linear regression may predict negative counts", html = TRUE, class = c("html", 
    "character")), type = "literal"), class = c("tutorial_question_answer", 
"tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
"character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
"character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
"character")), try_again = structure("Incorrect", html = TRUE, class = c("html", 
"character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
"character")), message = NULL, post_message = NULL), ids = list(
    answer = "revision_2-answer", question = "revision_2"), loading = NULL, 
    random_answer_order = FALSE, allow_retry = TRUE, seed = 697002699.675433, 
    options = list()), class = c("learnr_radio", "tutorial_question"
)), session = session)
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_radio", label = "revision_3", question = structure("Linear regression is unsuitable for modelling count outcomes since the variance is likely to increase with the mean and the errors will not be normally distributed", html = TRUE, class = c("html", 
"character")), answers = list(structure(list(id = "lnr_ans_5d02970", 
    option = "True", value = "True", label = structure("True", html = TRUE, class = c("html", 
    "character")), correct = TRUE, message = structure("Yes. With count outcomes, the errors will not be normally distributed but linear regression assumes that they are. This happens when the variance increases with the mean. In contrast, count regression is suitable to use when the variance increases with the mean (e.g., when variance = mean)", html = TRUE, class = c("html", 
    "character")), type = "literal"), class = c("tutorial_question_answer", 
"tutorial_quiz_answer")), structure(list(id = "lnr_ans_5755cd9", 
    option = "False", value = "False", label = structure("False", html = TRUE, class = c("html", 
    "character")), correct = FALSE, message = structure("No. With count outcomes, the errors will not be normally distributed but linear regression assumes that they are. This happens when the variance increases with the mean. In contrast, count regression is suitable to use when the variance increases with the mean (e.g., when variance = mean)", html = TRUE, class = c("html", 
    "character")), type = "literal"), class = c("tutorial_question_answer", 
"tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
"character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
"character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
"character")), try_again = structure("Incorrect", html = TRUE, class = c("html", 
"character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
"character")), message = NULL, post_message = NULL), ids = list(
    answer = "revision_3-answer", question = "revision_3"), loading = NULL, 
    random_answer_order = FALSE, allow_retry = TRUE, seed = 108433121.949507, 
    options = list()), class = c("learnr_radio", "tutorial_question"
)), session = session)
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-view_1-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-view_1-code-editor`)), session)
output$`tutorial-exercise-view_1-output` <- renderUI({
  `tutorial-exercise-view_1-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "view_1", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "view_1", code = "\nfor_shiny", opts = list(
        label = "\"view_1\"", exercise = "TRUE"), engine = "r")), 
    code_check = NULL, error_check = NULL, check = NULL, solution = NULL, 
    tests = NULL, options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "main_cache/html/", cache.vars = NULL, cache.lazy = TRUE, 
        dependson = NULL, autodep = FALSE, cache.rebuild = FALSE, 
        fig.keep = "high", fig.show = "asis", fig.align = "default", 
        fig.path = "main_files/figure-html/", dev = "png", dev.args = NULL, 
        dpi = 192, fig.ext = "png", fig.width = 6.5, fig.height = 4, 
        fig.env = "figure", fig.cap = NULL, fig.scap = NULL, 
        fig.lp = "fig:", fig.subcap = NULL, fig.pos = "", out.width = 624, 
        out.height = NULL, out.extra = NULL, fig.retina = 2, 
        external = TRUE, sanitize = FALSE, interval = 1, aniopts = "controls,loop", 
        warning = TRUE, error = TRUE, message = TRUE, render = NULL, 
        ref.label = NULL, child = NULL, engine = "r", split = FALSE, 
        include = TRUE, purl = TRUE, max.print = 1000, exercise.timelimit = 60, 
        exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "view_1", exercise = TRUE, code = c("", "for_shiny"
        ), out.width.px = 624, out.height.px = 384, params.src = "view_1, exercise = TRUE", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-names_1-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-names_1-code-editor`)), session)
output$`tutorial-exercise-names_1-output` <- renderUI({
  `tutorial-exercise-names_1-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "names_1", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "names_1", code = "for_shiny |> names()", opts = list(
        label = "\"names_1\"", exercise = "TRUE"), engine = "r")), 
    code_check = NULL, error_check = NULL, check = NULL, solution = NULL, 
    tests = NULL, options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "main_cache/html/", cache.vars = NULL, cache.lazy = TRUE, 
        dependson = NULL, autodep = FALSE, cache.rebuild = FALSE, 
        fig.keep = "high", fig.show = "asis", fig.align = "default", 
        fig.path = "main_files/figure-html/", dev = "png", dev.args = NULL, 
        dpi = 192, fig.ext = "png", fig.width = 6.5, fig.height = 4, 
        fig.env = "figure", fig.cap = NULL, fig.scap = NULL, 
        fig.lp = "fig:", fig.subcap = NULL, fig.pos = "", out.width = 624, 
        out.height = NULL, out.extra = NULL, fig.retina = 2, 
        external = TRUE, sanitize = FALSE, interval = 1, aniopts = "controls,loop", 
        warning = TRUE, error = TRUE, message = TRUE, render = NULL, 
        ref.label = NULL, child = NULL, engine = "r", split = FALSE, 
        include = TRUE, purl = TRUE, max.print = 1000, exercise.timelimit = 60, 
        exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "names_1", exercise = TRUE, code = "for_shiny |> names()", 
        out.width.px = 624, out.height.px = 384, params.src = "names_1, exercise = TRUE", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-rename_1-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-rename_1-code-editor`)), session)
output$`tutorial-exercise-rename_1-output` <- renderUI({
  `tutorial-exercise-rename_1-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "rename_1", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "rename_1", code = "df <- for_shiny # Rename dataframe\n\n# Rename variables\nnames(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",\n               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",\n               \"nzdep_decile\", \"arf_rate\")\ndf |> names() # Check that the variables have been renamed", 
        opts = list(label = "\"rename_1\"", exercise = "TRUE"), 
        engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "main_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "main_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = TRUE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "rename_1", exercise = TRUE, code = c("df <- for_shiny # Rename dataframe", 
        "", "# Rename variables", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
        "               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
        "               \"nzdep_decile\", \"arf_rate\")", "df |> names() # Check that the variables have been renamed"
        ), out.width.px = 624, out.height.px = 384, params.src = "rename_1, exercise = TRUE", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-duplicate_1-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-duplicate_1-code-editor`)), session)
output$`tutorial-exercise-duplicate_1-output` <- renderUI({
  `tutorial-exercise-duplicate_1-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "duplicate_1", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "duplicate_1", code = "stopifnot(df[df$Suburb |> duplicated(),] == 0)", 
        opts = list(label = "\"duplicate_1\"", exercise = "TRUE"), 
        engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "main_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "main_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = TRUE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "duplicate_1", exercise = TRUE, code = "stopifnot(df[df$Suburb |> duplicated(),] == 0)", 
        out.width.px = 624, out.height.px = 384, params.src = "duplicate_1, exercise = TRUE", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-range_1-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-range_1-code-editor`)), session)
output$`tutorial-exercise-range_1-output` <- renderUI({
  `tutorial-exercise-range_1-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "range_1", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "range_1", code = "df |> skim()", opts = list(
        label = "\"range_1\"", exercise = "TRUE"), engine = "r")), 
    code_check = NULL, error_check = NULL, check = NULL, solution = NULL, 
    tests = NULL, options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "main_cache/html/", cache.vars = NULL, cache.lazy = TRUE, 
        dependson = NULL, autodep = FALSE, cache.rebuild = FALSE, 
        fig.keep = "high", fig.show = "asis", fig.align = "default", 
        fig.path = "main_files/figure-html/", dev = "png", dev.args = NULL, 
        dpi = 192, fig.ext = "png", fig.width = 6.5, fig.height = 4, 
        fig.env = "figure", fig.cap = NULL, fig.scap = NULL, 
        fig.lp = "fig:", fig.subcap = NULL, fig.pos = "", out.width = 624, 
        out.height = NULL, out.extra = NULL, fig.retina = 2, 
        external = TRUE, sanitize = FALSE, interval = 1, aniopts = "controls,loop", 
        warning = TRUE, error = TRUE, message = TRUE, render = NULL, 
        ref.label = NULL, child = NULL, engine = "r", split = FALSE, 
        include = TRUE, purl = TRUE, max.print = 1000, exercise.timelimit = 60, 
        exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "range_1", exercise = TRUE, code = "df |> skim()", 
        out.width.px = 624, out.height.px = 384, params.src = "range_1, exercise = TRUE", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-variance_1-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-variance_1-code-editor`)), session)
output$`tutorial-exercise-variance_1-output` <- renderUI({
  `tutorial-exercise-variance_1-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "variance_1", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "variance_1", code = "\n\n", opts = list(label = "\"variance_1\"", 
        exercise = "TRUE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = structure(c("grade_this( {", 
    "if(isTRUE(all.equal(.solution, .result, tolerance = 0.1) )){", 
    "    pass(\"Yes, standard error = 1.17/ sqrt(413) = 0.0576\")", 
    "  }", "  fail(\"Please try again\")", "})"), chunk_opts = list(
        label = "variance_1-check")), solution = structure("1.17/sqrt(413)", chunk_opts = list(
        label = "variance_1-solution")), tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "main_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "main_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = TRUE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "variance_1", exercise = TRUE, out.width.px = 624, 
        out.height.px = 384, params.src = "variance_1, exercise = TRUE", 
        fig.num = 0L, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-variance_2-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-variance_2-code-editor`)), session)
output$`tutorial-exercise-variance_2-output` <- renderUI({
  `tutorial-exercise-variance_2-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "variance_2", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "variance_2", code = "\n\n", opts = list(label = "\"variance_2\"", 
        exercise = "TRUE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = structure(c("grade_this( {", 
    "if(isTRUE(all.equal(.solution, .result, tolerance = 0.1) )){", 
    "    pass(\"Yes, variance = (standard error)^2\")", "  }", 
    "  fail(\"Please try again\")", "})"), chunk_opts = list(
        label = "variance_2-check")), solution = structure("(1.17/sqrt(413))^2", chunk_opts = list(
        label = "variance_2-solution")), tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "main_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "main_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = TRUE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "variance_2", exercise = TRUE, out.width.px = 624, 
        out.height.px = 384, params.src = "variance_2, exercise = TRUE", 
        fig.num = 0L, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-missing_1-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-missing_1-code-editor`)), session)
output$`tutorial-exercise-missing_1-output` <- renderUI({
  `tutorial-exercise-missing_1-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "missing_1", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "missing_1", code = "df |> vis_miss(sort_miss = TRUE)", 
        opts = list(label = "\"missing_1\"", exercise = "TRUE"), 
        engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "main_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "main_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = TRUE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "missing_1", exercise = TRUE, code = "df |> vis_miss(sort_miss = TRUE)", 
        out.width.px = 624, out.height.px = 384, params.src = "missing_1, exercise = TRUE", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-histogram_1-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-histogram_1-code-editor`)), session)
output$`tutorial-exercise-histogram_1-output` <- renderUI({
  `tutorial-exercise-histogram_1-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "histogram_1", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "histogram_1", code = "df$arf_count |> histogram()", 
        opts = list(label = "\"histogram_1\"", exercise = "TRUE"), 
        engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "main_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "main_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = TRUE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "histogram_1", exercise = TRUE, code = "df$arf_count |> histogram()", 
        out.width.px = 624, out.height.px = 384, params.src = "histogram_1, exercise = TRUE", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-histogram_2-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-histogram_2-code-editor`)), session)
output$`tutorial-exercise-histogram_2-output` <- renderUI({
  `tutorial-exercise-histogram_2-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "histogram_2", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "histogram_2", code = "df$arf_rate |> histogram()", 
        opts = list(label = "\"histogram_2\"", exercise = "TRUE"), 
        engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "main_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "main_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = TRUE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "histogram_2", exercise = TRUE, code = "df$arf_rate |> histogram()", 
        out.width.px = 624, out.height.px = 384, params.src = "histogram_2, exercise = TRUE", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_radio", label = "histogram_3", 
    question = structure("<p>To decide what type of regression to use for our models, we need to check for overdispersion.<br />\nLook at your first histogram above – for <code>arf_count<\u002fcode>. In the previous section, you estimated the mean of  <code>arf_count<\u002fcode> displayed from using the <code>skim<\u002fcode> function (<code>0.55<\u002fcode>) and compare it with the variance you calculated (<code>0.0033<\u002fcode>).<\u002fp>\n<p>Which of the following, if any, is consistent with underdispersion?<\u002fp>\n", html = TRUE, class = c("html", 
    "character")), answers = list(structure(list(id = "lnr_ans_c94dbec", 
        option = "Variance < mean", value = "Variance < mean", 
        label = structure("Variance &lt; mean", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = structure("Yes. The variance (0.0033) is substantially smaller than the mean (0.55).", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_d0080ec", 
        option = "Long tail of the distribution only", value = "Long tail of the distribution only", 
        label = structure("Long tail of the distribution only", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = structure("No. A long tail is a feature of a power distribution. Please try again", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
    "character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
    "character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
    "character")), try_again = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), message = NULL, post_message = NULL), ids = list(
        answer = "histogram_3-answer", question = "histogram_3"), 
    loading = NULL, random_answer_order = FALSE, allow_retry = TRUE, 
    seed = 818099589.619043, options = list()), class = c("learnr_radio", 
"tutorial_question")), session = session)
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_radio", label = "histogram_4", 
    question = structure("Your results should point to evidence of underdispersion. Given this observation, what type of regression would improve model fit?", html = TRUE, class = c("html", 
    "character")), answers = list(structure(list(id = "lnr_ans_bfc9ce", 
        option = "Linear", value = "Linear", label = structure("Linear", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = structure("The outcome, ARF count, comprises whole numbers, even if there is overdispersion. So linear regression is unsuitable. Please try again", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_2509dcf", 
        option = "Poisson", value = "Poisson", label = structure("Poisson", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = structure("No, as there is overdispersion, we need to account for this by modelling a variant of Poisson regression. Please try again", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_68423b2", 
        option = "Quasi-Poisson regression or negative-binomial", 
        value = "Quasi-Poisson regression or negative-binomial", 
        label = structure("Quasi-Poisson regression or negative-binomial", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = structure("Yes, these forms of regression are used for accounting for over or under dispersion when modelling outcomes comprising counts", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_4d5cd9f", 
        option = "Logistic regression", value = "Logistic regression", 
        label = structure("Logistic regression", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = structure("This is used when the outcome is binary (2 values only), but our outcome (ARF count) comprises positive integers with more than 2 values. Please try again", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
    "character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
    "character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
    "character")), try_again = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), message = NULL, post_message = NULL), ids = list(
        answer = "histogram_4-answer", question = "histogram_4"), 
    loading = NULL, random_answer_order = FALSE, allow_retry = TRUE, 
    seed = 1685178221.21528, options = list()), class = c("learnr_radio", 
"tutorial_question")), session = session)
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-histogram_5-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-histogram_5-code-editor`)), session)
output$`tutorial-exercise-histogram_5-output` <- renderUI({
  `tutorial-exercise-histogram_5-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "histogram_5", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "histogram_5", code = "df$Permethrin_rate |> histogram()", 
        opts = list(label = "\"histogram_5\"", exercise = "TRUE"), 
        engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "main_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "main_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = TRUE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "histogram_5", exercise = TRUE, code = "df$Permethrin_rate |> histogram()", 
        out.width.px = 624, out.height.px = 384, params.src = "histogram_5, exercise = TRUE", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-histogram_6-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-histogram_6-code-editor`)), session)
output$`tutorial-exercise-histogram_6-output` <- renderUI({
  `tutorial-exercise-histogram_6-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "histogram_6", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "histogram_6", code = "df$Pacific_proportion \ndf$Maori_proportion\ndf$nzdep_decile ", 
        opts = list(label = "\"histogram_6\"", exercise = "TRUE"), 
        engine = "r")), code_check = NULL, error_check = NULL, 
    check = structure("grade_this_code(\"Good, you can now visualise the distribution of continuous and count variables with histograms!\")", chunk_opts = list(
        label = "histogram_6-check")), solution = structure(c("df$Pacific_proportion |> histogram()", 
    "df$Maori_proportion |> histogram()", "df$nzdep_decile |> histogram()"
    ), chunk_opts = list(label = "histogram_6-solution")), tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "main_cache/html/", cache.vars = NULL, cache.lazy = TRUE, 
        dependson = NULL, autodep = FALSE, cache.rebuild = FALSE, 
        fig.keep = "high", fig.show = "asis", fig.align = "default", 
        fig.path = "main_files/figure-html/", dev = "png", dev.args = NULL, 
        dpi = 192, fig.ext = "png", fig.width = 6.5, fig.height = 4, 
        fig.env = "figure", fig.cap = NULL, fig.scap = NULL, 
        fig.lp = "fig:", fig.subcap = NULL, fig.pos = "", out.width = 624, 
        out.height = NULL, out.extra = NULL, fig.retina = 2, 
        external = TRUE, sanitize = FALSE, interval = 1, aniopts = "controls,loop", 
        warning = TRUE, error = TRUE, message = TRUE, render = NULL, 
        ref.label = NULL, child = NULL, engine = "r", split = FALSE, 
        include = TRUE, purl = TRUE, max.print = 1000, exercise.timelimit = 60, 
        exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "histogram_6", exercise = TRUE, code = c("df$Pacific_proportion ", 
        "df$Maori_proportion", "df$nzdep_decile "), out.width.px = 624, 
        out.height.px = 384, params.src = "histogram_6, exercise = TRUE", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_radio", label = "histogram_7", 
    question = structure("How would you interpret the histogram for <code>Pacific_proportion<\u002fcode>?", html = TRUE, class = c("html", 
    "character")), answers = list(structure(list(id = "lnr_ans_4284d5", 
        option = "The proportion of Pacific people is evenly distributed in suburbs in Auckland", 
        value = "The proportion of Pacific people is evenly distributed in suburbs in Auckland", 
        label = structure("The proportion of Pacific people is evenly distributed in suburbs in Auckland", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = structure("No, in fact, you can see that it is very uneven! Please try again", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_1e31f50", 
        option = "Most suburbs in Auckland have a high proportion of Pacific people living in them", 
        value = "Most suburbs in Auckland have a high proportion of Pacific people living in them", 
        label = structure("Most suburbs in Auckland have a high proportion of Pacific people living in them", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = structure("No, please reconsider the plot and try again. Please try again", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_bbd59f5", 
        option = "A small number of suburbs in Auckland have a very high proportion of Pacific people living in them", 
        value = "A small number of suburbs in Auckland have a very high proportion of Pacific people living in them", 
        label = structure("A small number of suburbs in Auckland have a very high proportion of Pacific people living in them", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = structure("Yes, this is what the histogram shows.", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
    "character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
    "character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
    "character")), try_again = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), message = NULL, post_message = NULL), ids = list(
        answer = "histogram_7-answer", question = "histogram_7"), 
    loading = NULL, random_answer_order = FALSE, allow_retry = TRUE, 
    seed = 321393580.85034, options = list()), class = c("learnr_radio", 
"tutorial_question")), session = session)
</script>


<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-histogram_8-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-histogram_8-code-editor`)), session)
output$`tutorial-exercise-histogram_8-output` <- renderUI({
  `tutorial-exercise-histogram_8-result`()
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "histogram_8", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "histogram_8", code = "# 5 groups instead of 10\ndf$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),\n                     labels = c(\"1 to 2 (least deprived)\",\n                                \"3 to 4\",\n                                \"5 to 6\",\n                                \"7 to 8\",\n                                \"9 to 10 (most deprived)\"))\n\n# Check the frequency of our 5 groups\nepiDisplay::tabpct(df$nzDepQuint, df$nzdep_decile)", 
        opts = list(label = "\"histogram_8\"", exercise = "TRUE"), 
        engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "main_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "main_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = TRUE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "histogram_8", exercise = TRUE, code = c("# 5 groups instead of 10", 
        "df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
        "                     labels = c(\"1 to 2 (least deprived)\",", 
        "                                \"3 to 4\",", "                                \"5 to 6\",", 
        "                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
        "", "# Check the frequency of our 5 groups", "epiDisplay::tabpct(df$nzDepQuint, df$nzdep_decile)"
        ), out.width.px = 624, out.height.px = 384, params.src = "histogram_8, exercise = TRUE", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>


<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-scatterplot_1-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-scatterplot_1-code-editor`)), session)
output$`tutorial-exercise-scatterplot_1-output` <- renderUI({
  `tutorial-exercise-scatterplot_1-result`()
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "scatterplot_1", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "scatterplot_1", code = "car::scatterplot(arf_rate ~ Permethrin_rate, data = df)", 
        opts = list(label = "\"scatterplot_1\"", exercise = "TRUE"), 
        engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "main_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "main_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = TRUE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "scatterplot_1", exercise = TRUE, code = "car::scatterplot(arf_rate ~ Permethrin_rate, data = df)", 
        out.width.px = 624, out.height.px = 384, params.src = "scatterplot_1, exercise = TRUE", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_radio", label = "scatter_1", question = structure("How would you interpret the scatter plot between <code>Permethrin_rate<\u002fcode> dispensing and <code>arf_rate<\u002fcode>?", html = TRUE, class = c("html", 
"character")), answers = list(structure(list(id = "lnr_ans_49bdbb9", 
    option = "No evidence of association", value = "No evidence of association", 
    label = structure("No evidence of association", html = TRUE, class = c("html", 
    "character")), correct = FALSE, message = structure("No, this would be expected if there were a horizontal regression line", html = TRUE, class = c("html", 
    "character")), type = "literal"), class = c("tutorial_question_answer", 
"tutorial_quiz_answer")), structure(list(id = "lnr_ans_1f7af22", 
    option = "Negative association", value = "Negative association", 
    label = structure("Negative association", html = TRUE, class = c("html", 
    "character")), correct = FALSE, message = structure("No, we would expect to see a line in the top right hand corner going to the bottom left.", html = TRUE, class = c("html", 
    "character")), type = "literal"), class = c("tutorial_question_answer", 
"tutorial_quiz_answer")), structure(list(id = "lnr_ans_ae5a170", 
    option = "A positive association", value = "A positive association", 
    label = structure("A positive association", html = TRUE, class = c("html", 
    "character")), correct = TRUE, message = structure("Yes, the greater the permethrin dispensing, the greater the rate of rheumatic fever", html = TRUE, class = c("html", 
    "character")), type = "literal"), class = c("tutorial_question_answer", 
"tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
"character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
"character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
"character")), try_again = structure("Incorrect", html = TRUE, class = c("html", 
"character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
"character")), message = NULL, post_message = NULL), ids = list(
    answer = "scatter_1-answer", question = "scatter_1"), loading = NULL, 
    random_answer_order = FALSE, allow_retry = TRUE, seed = 2139727399.50361, 
    options = list()), class = c("learnr_radio", "tutorial_question"
)), session = session)
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-scatterplot_2-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-scatterplot_2-code-editor`)), session)
output$`tutorial-exercise-scatterplot_2-output` <- renderUI({
  `tutorial-exercise-scatterplot_2-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "scatterplot_2", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "scatterplot_2", code = "arf_rate ~ nzdep_decile\narf_rate ~ Pacific_proportion \narf_rate ~ Maori_proportion ", 
        opts = list(label = "\"scatterplot_2\"", exercise = "TRUE"), 
        engine = "r")), code_check = NULL, error_check = NULL, 
    check = structure("grade_this_code(\"Good! You can now visualise the distribution of two continuous variables!\")", chunk_opts = list(
        label = "scatterplot_2-check")), solution = structure(c("car::scatterplot(arf_rate ~ nzdep_decile, data = df)", 
    "car::scatterplot(arf_rate ~ Pacific_proportion, data = df)", 
    "car::scatterplot(arf_rate ~ Maori_proportion, data = df)"
    ), chunk_opts = list(label = "scatterplot_2-solution")), 
    tests = NULL, options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "main_cache/html/", cache.vars = NULL, cache.lazy = TRUE, 
        dependson = NULL, autodep = FALSE, cache.rebuild = FALSE, 
        fig.keep = "high", fig.show = "asis", fig.align = "default", 
        fig.path = "main_files/figure-html/", dev = "png", dev.args = NULL, 
        dpi = 192, fig.ext = "png", fig.width = 6.5, fig.height = 4, 
        fig.env = "figure", fig.cap = NULL, fig.scap = NULL, 
        fig.lp = "fig:", fig.subcap = NULL, fig.pos = "", out.width = 624, 
        out.height = NULL, out.extra = NULL, fig.retina = 2, 
        external = TRUE, sanitize = FALSE, interval = 1, aniopts = "controls,loop", 
        warning = TRUE, error = TRUE, message = TRUE, render = NULL, 
        ref.label = NULL, child = NULL, engine = "r", split = FALSE, 
        include = TRUE, purl = TRUE, max.print = 1000, exercise.timelimit = 60, 
        exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "scatterplot_2", exercise = TRUE, code = c("arf_rate ~ nzdep_decile", 
        "arf_rate ~ Pacific_proportion ", "arf_rate ~ Maori_proportion "
        ), out.width.px = 624, out.height.px = 384, params.src = "scatterplot_2, exercise = TRUE", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_checkbox", label = "scatter_2_int", 
    question = structure("How would you interpret the scatter plot between the potential confounders and <code>arf_rate<\u002fcode>?", html = TRUE, class = c("html", 
    "character")), answers = list(structure(list(id = "lnr_ans_55065fc", 
        option = "No evidence of association in any variables", 
        value = "No evidence of association in any variables", 
        label = structure("No evidence of association in any variables", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = structure("No, this would be expected if the regression lines were horizontal", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_4fcbe2f", 
        option = "All Negative associations", value = "All Negative associations", 
        label = structure("All Negative associations", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = structure("No, we would expect to see lines in the top right hand corner going to the bottom left.", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_94a73eb", 
        option = "All positive associations", value = "All positive associations", 
        label = structure("All positive associations", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = structure("Yes, the greater the permethrin dispensing, the greater the rate of rheumatic fever", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_afdde26", 
        option = "Something else", value = "Something else", 
        label = structure("Something else", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = structure("No, please try again!", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
    "character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
    "character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
    "character")), try_again = structure("Incorrect. Be sure to select every correct answer.", html = TRUE, class = c("html", 
    "character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), message = NULL, post_message = NULL), ids = list(
        answer = "scatter_2_int-answer", question = "scatter_2_int"), 
    loading = NULL, random_answer_order = FALSE, allow_retry = TRUE, 
    seed = 523419333.756264, options = list()), class = c("learnr_checkbox", 
"tutorial_question")), session = session)
</script>


<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-scatterplot_4-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-scatterplot_4-code-editor`)), session)
output$`tutorial-exercise-scatterplot_4-output` <- renderUI({
  `tutorial-exercise-scatterplot_4-result`()
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "scatterplot_4", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "scatterplot_4", code = "## Select variables of interest into new data.frame\nmy_data <- df[, c(\"Pacific_proportion\", \"Maori_proportion\", \"Permethrin_rate\", \"nzdep_decile\", \"arf_rate\")]\n\n## Chart the correlations and regressions and distributions of each variable.\nPerformanceAnalytics::chart.Correlation(my_data, histogram = TRUE, pch = 19)", 
        opts = list(label = "\"scatterplot_4\"", exercise = "TRUE"), 
        engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "main_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "main_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = TRUE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "scatterplot_4", exercise = TRUE, code = c("## Select variables of interest into new data.frame", 
        "my_data <- df[, c(\"Pacific_proportion\", \"Maori_proportion\", \"Permethrin_rate\", \"nzdep_decile\", \"arf_rate\")]", 
        "", "## Chart the correlations and regressions and distributions of each variable.", 
        "PerformanceAnalytics::chart.Correlation(my_data, histogram = TRUE, pch = 19)"
        ), out.width.px = 624, out.height.px = 384, params.src = "scatterplot_4, exercise = TRUE", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_radio", label = "scatterplot_5", 
    question = structure("What is the Pearson correlation coefficient (<code>r<\u002fcode>) for the association between Pacific proportion and permethrin rate?", html = TRUE, class = c("html", 
    "character")), answers = list(structure(list(id = "lnr_ans_f792b42", 
        option = "`r` = 0.55", value = "`r` = 0.55", label = structure("<code>r<\u002fcode> = 0.55", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = structure("No, <code>r<\u002fcode> = 0.55 for the correlation between <code>Pacific_proportion<\u002fcode> and <code>Maori_proportion<\u002fcode>", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_a45b752", 
        option = "`r` = 0.66", value = "`r` = 0.66", label = structure("<code>r<\u002fcode> = 0.66", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = structure("No, <code>r<\u002fcode> = 0.66 for the correlations of <code>Permethrin_rate<\u002fcode> with <code>Maori_proportion<\u002fcode> and <code>arf_rate<\u002fcode>", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_859ba4b", 
        option = "`r` = 0.88", value = "`r` = 0.88", label = structure("<code>r<\u002fcode> = 0.88", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = structure("Yes, the row for <code>Pacific_proportion<\u002fcode> intersects with the column for <code>Permethrin_rate<\u002fcode> at the cell that has 0.88 in it. A correlation of 0.88 is extremely high and suggests a very strong association.", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_87139e3", 
        option = "`r` = 0.79", value = "`r` = 0.79", label = structure("<code>r<\u002fcode> = 0.79", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = structure("No, <code>r<\u002fcode> = 0.79 for the correlation between <code>Permethrin_rate<\u002fcode> and <code>nzdep_decile<\u002fcode>", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
    "character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
    "character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
    "character")), try_again = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), message = NULL, post_message = NULL), ids = list(
        answer = "scatterplot_5-answer", question = "scatterplot_5"), 
    loading = NULL, random_answer_order = FALSE, allow_retry = TRUE, 
    seed = 1723787224.1973, options = list()), class = c("learnr_radio", 
"tutorial_question")), session = session)
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-regression_1-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-regression_1-code-editor`)), session)
output$`tutorial-exercise-regression_1-output` <- renderUI({
  `tutorial-exercise-regression_1-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "regression_1", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "regression_1", code = "## Make offset variable\ndf$log_population <- log(df$population)\n\n## Build model\nfit <- glm(arf_count ~ Permethrin_rate +\n           nzDepQuint +\n           Pacific_proportion +\n           Maori_proportion, \n           offset = log_population,\n           family = poisson(link = \"log\"),\n           data = df)\n\n## Print summary of model\nsummary(fit)", 
        opts = list(label = "\"regression_1\"", exercise = "TRUE"), 
        engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "main_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "main_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = TRUE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "regression_1", exercise = TRUE, code = c("## Make offset variable", 
        "df$log_population <- log(df$population)", "", "## Build model", 
        "fit <- glm(arf_count ~ Permethrin_rate +", "           nzDepQuint +", 
        "           Pacific_proportion +", "           Maori_proportion, ", 
        "           offset = log_population,", "           family = poisson(link = \"log\"),", 
        "           data = df)", "", "## Print summary of model", 
        "summary(fit)"), out.width.px = 624, out.height.px = 384, 
        params.src = "regression_1, exercise = TRUE", fig.num = 0, 
        exercise.df_print = "paged"), engine = "r", version = "4"), class = c("r", 
"tutorial_exercise")))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_checkbox", label = "regression_2", 
    question = structure("Commonly, offsets are based on the time spent observing each outcome (observation period). When would it be appropriate to include <code>log(time in years)<\u002fcode> as an offset? (select all that apply)", html = TRUE, class = c("html", 
    "character")), answers = list(structure(list(id = "lnr_ans_c8e96b8", 
        option = "If the observation period differed by suburb", 
        value = "If the observation period differed by suburb", 
        label = structure("If the observation period differed by suburb", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = structure("First, if the observation period differed by suburb, then  larger numbers of cases would be likely to in suburbs observed for longer periods. Having an offset based on time accounts for this.", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_11eeff0", 
        option = "If we wanted our outcome to be rates based on counts per *year*", 
        value = "If we wanted our outcome to be rates based on counts per *year*", 
        label = structure("If we wanted our outcome to be rates based on counts per <em>year<\u002fem>", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = structure("Second, with counts (<code>arf_count<\u002fcode>) as the outcome, adding <code>log(time)<\u002fcode> as the offset makes the model represent a rate in counts per <em>year<\u002fem>.", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_b31738", 
        option = "If we had data on the observation period of each suburb", 
        value = "If we had data on the observation period of each suburb", 
        label = structure("If we had data on the observation period of each suburb", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = structure("Third, we need the observation period for each suburb.", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_7dba47c", 
        option = "None of the above", value = "None of the above", 
        label = structure("None of the above", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = structure("Please try again", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
    "character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
    "character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
    "character")), try_again = structure("Incorrect. Be sure to select every correct answer.", html = TRUE, class = c("html", 
    "character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), message = NULL, post_message = NULL), ids = list(
        answer = "regression_2-answer", question = "regression_2"), 
    loading = NULL, random_answer_order = FALSE, allow_retry = TRUE, 
    seed = 449757891.290565, options = list()), class = c("learnr_checkbox", 
"tutorial_question")), session = session)
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_radio", label = "regression_3", 
    question = structure("For <code>Permethrin_rate<\u002fcode>, you should get an <code>Estimate<\u002fcode> value of 3.527e-04. What does this tell us?", html = TRUE, class = c("html", 
    "character")), answers = list(structure(list(id = "lnr_ans_e9d10b", 
        option = "There is, on average, 3.527e-04 = 0.0003527 more acute rheumatic fever for every one unit increase in permethrin rate", 
        value = "There is, on average, 3.527e-04 = 0.0003527 more acute rheumatic fever for every one unit increase in permethrin rate", 
        label = structure("There is, on average, 3.527e-04 = 0.0003527 more acute rheumatic fever for every one unit increase in permethrin rate", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = structure("No. 3.527e-04 is measured on a natural-log scale. So you need to exponentiate it to give a meaningful interpretation", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_d8dd176", 
        option = "The incidence rate of acute rheumatic fever is, on average, 3.527e-04 = 0.0003527 times higher for every 1-unit increase in permethrin rate", 
        value = "The incidence rate of acute rheumatic fever is, on average, 3.527e-04 = 0.0003527 times higher for every 1-unit increase in permethrin rate", 
        label = structure("The incidence rate of acute rheumatic fever is, on average, 3.527e-04 = 0.0003527 times higher for every 1-unit increase in permethrin rate", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = structure("No. 3.527e-04 is measured on a natural-log scale. So you need to exponentiate it to interpret the figure.", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_4f8c910", 
        option = "There is, on average, exp(3.527e-04) more acute rheumatic fever for every 1-unit increase in permethrin rate", 
        value = "There is, on average, exp(3.527e-04) more acute rheumatic fever for every 1-unit increase in permethrin rate", 
        label = structure("There is, on average, exp(3.527e-04) more acute rheumatic fever for every 1-unit increase in permethrin rate", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = structure("No. Exponentiating this converts it into an incidence rate ratio. However, <em>exp(3.527e-04) more<\u002fem> does not imply a ratio. What word do we need to add to imply a ratio?", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_58ad618", 
        option = "There is, on average, exp(3.527e-04) = 1.00035 times more acute rheumatic fever for every 1-unit increase in permethrin dispensing rate", 
        value = "There is, on average, exp(3.527e-04) = 1.00035 times more acute rheumatic fever for every 1-unit increase in permethrin dispensing rate", 
        label = structure("There is, on average, exp(3.527e-04) = 1.00035 times more acute rheumatic fever for every 1-unit increase in permethrin dispensing rate", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = structure("Yes, exponentiating 3.527e-04 converts it into an incidence rate ratio. The word <em>times<\u002fem> suggests a ratio", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
    "character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
    "character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
    "character")), try_again = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), message = NULL, post_message = NULL), ids = list(
        answer = "regression_3-answer", question = "regression_3"), 
    loading = NULL, random_answer_order = FALSE, allow_retry = TRUE, 
    seed = 858705289.100134, options = list()), class = c("learnr_radio", 
"tutorial_question")), session = session)
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-regression_4-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-regression_4-code-editor`)), session)
output$`tutorial-exercise-regression_4-output` <- renderUI({
  `tutorial-exercise-regression_4-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "regression_4", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "regression_4", code = "jtools::summ(fit, exp = TRUE)", 
        opts = list(label = "\"regression_4\"", exercise = "TRUE"), 
        engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "main_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "main_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = TRUE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "regression_4", exercise = TRUE, code = "jtools::summ(fit, exp = TRUE)", 
        out.width.px = 624, out.height.px = 384, params.src = "regression_4, exercise = TRUE", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_radio", label = "regression_5", 
    question = structure("For <code>Pacific_proportion<\u002fcode>, you get an <code>exp(Est.)<\u002fcode> value of 11.56. What does this tell us?", html = TRUE, class = c("html", 
    "character")), answers = list(structure(list(id = "lnr_ans_979668c", 
        option = "There is, on average, 11.56 more acute rheumatic fever for every 1-unit increase in Pacific proportion", 
        value = "There is, on average, 11.56 more acute rheumatic fever for every 1-unit increase in Pacific proportion", 
        label = structure("There is, on average, 11.56 more acute rheumatic fever for every 1-unit increase in Pacific proportion", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = structure("No. 11.56 is a rate. What word do we need to add to imply a ratio?", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_55d76b1", 
        option = "The incidence of acute rheumatic fever is, on average, 11.56 times higher for every 1-unit increase in Pacific proportion", 
        value = "The incidence of acute rheumatic fever is, on average, 11.56 times higher for every 1-unit increase in Pacific proportion", 
        label = structure("The incidence of acute rheumatic fever is, on average, 11.56 times higher for every 1-unit increase in Pacific proportion", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = structure("Well done! <em>11.56 times<\u002fem> implies a ratio and 11.56 is the incidence rate ratio", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_3fdb4e8", 
        option = "There is, on average, `exp(11.56)` more acute rheumatic fever for every 1-unit increase in Pacific proportion", 
        value = "There is, on average, `exp(11.56)` more acute rheumatic fever for every 1-unit increase in Pacific proportion", 
        label = structure("There is, on average, <code>exp(11.56)<\u002fcode> more acute rheumatic fever for every 1-unit increase in Pacific proportion", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = structure("No, 11.56 is already an exponentiated value (R code above reads <code>exp = TRUE<\u002fcode>) – you don’t need to exponentiate again!", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_ab8fdad", 
        option = "The incidence rate of acute rheumatic fever is, on average, `exp(11.56)` times higher for every 1-unit increase in Pacific proportion", 
        value = "The incidence rate of acute rheumatic fever is, on average, `exp(11.56)` times higher for every 1-unit increase in Pacific proportion", 
        label = structure("The incidence rate of acute rheumatic fever is, on average, <code>exp(11.56)<\u002fcode> times higher for every 1-unit increase in Pacific proportion", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = structure("No, 11.56 is already an exponentiated value (R code above reads <code>exp = TRUE<\u002fcode>) – you don’t need to exponentiate again!", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
    "character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
    "character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
    "character")), try_again = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), message = NULL, post_message = NULL), ids = list(
        answer = "regression_5-answer", question = "regression_5"), 
    loading = NULL, random_answer_order = FALSE, allow_retry = TRUE, 
    seed = 1834907642.64555, options = list()), class = c("learnr_radio", 
"tutorial_question")), session = session)
</script>


<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-regression_6-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-regression_6-code-editor`)), session)
output$`tutorial-exercise-regression_6-output` <- renderUI({
  `tutorial-exercise-regression_6-result`()
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "regression_6", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "regression_6", code = "fit1 <- glm(arf_count ~ I(Permethrin_rate/1000) +\n             nzDepQuint +\n             Pacific_proportion +\n             Maori_proportion + offset(log(population)),\n           family = poisson(link = \"log\"),\n           data = df)\nsummary(fit1)", 
        opts = list(label = "\"regression_6\"", exercise = "TRUE"), 
        engine = "r")), code_check = NULL, error_check = NULL, 
    check = structure("grade_this_code(\"Good. It’s important to know how to change to a `quasipoisson` model when there is evidence of over or under dispersion\")", chunk_opts = list(
        label = "regression_6-check")), solution = structure(c("fit_qp <- glm(arf_count ~ I(Permethrin_rate/1000) +", 
    "             nzDepQuint +", "             Pacific_proportion +", 
    "             Maori_proportion + offset(log(population)),", 
    "           family = quasipoisson(link = \"log\"),", "           data = df)", 
    "summary(fit_qp)"), chunk_opts = list(label = "regression_6-solution")), 
    tests = NULL, options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "main_cache/html/", cache.vars = NULL, cache.lazy = TRUE, 
        dependson = NULL, autodep = FALSE, cache.rebuild = FALSE, 
        fig.keep = "high", fig.show = "asis", fig.align = "default", 
        fig.path = "main_files/figure-html/", dev = "png", dev.args = NULL, 
        dpi = 192, fig.ext = "png", fig.width = 6.5, fig.height = 4, 
        fig.env = "figure", fig.cap = NULL, fig.scap = NULL, 
        fig.lp = "fig:", fig.subcap = NULL, fig.pos = "", out.width = 624, 
        out.height = NULL, out.extra = NULL, fig.retina = 2, 
        external = TRUE, sanitize = FALSE, interval = 1, aniopts = "controls,loop", 
        warning = TRUE, error = TRUE, message = TRUE, render = NULL, 
        ref.label = NULL, child = NULL, engine = "r", split = FALSE, 
        include = TRUE, purl = TRUE, max.print = 1000, exercise.timelimit = 60, 
        exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "regression_6", exercise = TRUE, code = c("fit1 <- glm(arf_count ~ I(Permethrin_rate/1000) +", 
        "             nzDepQuint +", "             Pacific_proportion +", 
        "             Maori_proportion + offset(log(population)),", 
        "           family = poisson(link = \"log\"),", "           data = df)", 
        "summary(fit1)"), out.width.px = 624, out.height.px = 384, 
        params.src = "regression_6, exercise = TRUE", fig.num = 0, 
        exercise.df_print = "paged"), engine = "r", version = "4"), class = c("r", 
"tutorial_exercise")))
</script>


<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-regression_7-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-regression_7-code-editor`)), session)
output$`tutorial-exercise-regression_7-output` <- renderUI({
  `tutorial-exercise-regression_7-result`()
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "regression_7", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "regression_7", code = "# jtools::summ(model_object, exp = logical_value)", 
        opts = list(label = "\"regression_7\"", exercise = "TRUE"), 
        engine = "r")), code_check = NULL, error_check = NULL, 
    check = structure("grade_this_code(\"Great! Above, we created a model named `fit_qp`. As we want to exponentiate, the logical value for the `exp` argument will be `TRUE`\")", chunk_opts = list(
        label = "regression_7-check")), solution = structure("jtools::summ(fit_qp, exp = TRUE)", chunk_opts = list(
        label = "regression_7-solution")), tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "main_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "main_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = TRUE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "regression_7", exercise = TRUE, code = "# jtools::summ(model_object, exp = logical_value)", 
        out.width.px = 624, out.height.px = 384, params.src = "regression_7, exercise = TRUE", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>


<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-regression_8-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-regression_8-code-editor`)), session)
output$`tutorial-exercise-regression_8-output` <- renderUI({
  `tutorial-exercise-regression_8-result`()
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "regression_8", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "regression_8", code = "jtools::plot_summs(fit_qp, exp = TRUE)", 
        opts = list(label = "\"regression_8\"", exercise = "TRUE"), 
        engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "main_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "main_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = TRUE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "regression_8", exercise = TRUE, code = "jtools::plot_summs(fit_qp, exp = TRUE)", 
        out.width.px = 624, out.height.px = 384, params.src = "regression_8, exercise = TRUE", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>


<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-multicollinearity_1-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-multicollinearity_1-code-editor`)), session)
output$`tutorial-exercise-multicollinearity_1-output` <- renderUI({
  `tutorial-exercise-multicollinearity_1-result`()
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "multicollinearity_1", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "multicollinearity_1", code = "cor(df$Pacific_proportion, df$Permethrin_rate)", 
        opts = list(label = "\"multicollinearity_1\"", exercise = "TRUE"), 
        engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "main_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "main_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = TRUE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "multicollinearity_1", exercise = TRUE, code = "cor(df$Pacific_proportion, df$Permethrin_rate)", 
        out.width.px = 624, out.height.px = 384, params.src = "multicollinearity_1, exercise = TRUE", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>


<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-regression_9-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-regression_9-code-editor`)), session)
output$`tutorial-exercise-regression_9-output` <- renderUI({
  `tutorial-exercise-regression_9-result`()
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "regression_9", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "regression_9", code = "df$log_population <- log(df$population)\n\nfit3 <- glm(arf_count ~ I(Permethrin_rate/1000) +\n            nzDepQuint +\n             #Pacific_proportion +\n            Maori_proportion, offset=(log(population)),\n           family = quasipoisson(link = \"log\"),\n           data = df)\n\nsummary(fit3)", 
        opts = list(label = "\"regression_9\"", exercise = "TRUE"), 
        engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "main_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "main_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = TRUE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "regression_9", exercise = TRUE, code = c("df$log_population <- log(df$population)", 
        "", "fit3 <- glm(arf_count ~ I(Permethrin_rate/1000) +", 
        "            nzDepQuint +", "             #Pacific_proportion +", 
        "            Maori_proportion, offset=(log(population)),", 
        "           family = quasipoisson(link = \"log\"),", 
        "           data = df)", "", "summary(fit3)"), out.width.px = 624, 
        out.height.px = 384, params.src = "regression_9, exercise = TRUE", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>


<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-regression_10-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-regression_10-code-editor`)), session)
output$`tutorial-exercise-regression_10-output` <- renderUI({
  `tutorial-exercise-regression_10-result`()
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "regression_10", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "regression_10", code = "df$log_population <- log(df$population)\n\nfit4 <- glm(arf_count ~  I(Permethrin_rate/1000) +\n              nzDepQuint +\n              #Pacific_proportion +\n              Maori_proportion, offset=log_population,\n              family = quasipoisson(link = \"log\"),\n              data = df)\n\nsummary(fit4)", 
        opts = list(label = "\"regression_10\"", exercise = "TRUE"), 
        engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "main_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "main_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = TRUE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "regression_10", exercise = TRUE, code = c("df$log_population <- log(df$population)", 
        "", "fit4 <- glm(arf_count ~  I(Permethrin_rate/1000) +", 
        "              nzDepQuint +", "              #Pacific_proportion +", 
        "              Maori_proportion, offset=log_population,", 
        "              family = quasipoisson(link = \"log\"),", 
        "              data = df)", "", "summary(fit4)"), out.width.px = 624, 
        out.height.px = 384, params.src = "regression_10, exercise = TRUE", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>


<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-regression_11-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-regression_11-code-editor`)), session)
output$`tutorial-exercise-regression_11-output` <- renderUI({
  `tutorial-exercise-regression_11-result`()
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "regression_11", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "regression_11", code = "fit5 <- glm(arf_count ~ I(Permethrin_rate/1000) +\n              nzDepQuint +\n              Pacific_proportion +\n              Maori_proportion, offset(log(population)),\n              family = quasipoisson(link = \"log\"),\n              data = df)\n\nsummary(fit5)", 
        opts = list(label = "\"regression_11\"", exercise = "TRUE"), 
        engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "main_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "main_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = TRUE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "regression_11", exercise = TRUE, code = c("fit5 <- glm(arf_count ~ I(Permethrin_rate/1000) +", 
        "              nzDepQuint +", "              Pacific_proportion +", 
        "              Maori_proportion, offset(log(population)),", 
        "              family = quasipoisson(link = \"log\"),", 
        "              data = df)", "", "summary(fit5)"), out.width.px = 624, 
        out.height.px = 384, params.src = "regression_11, exercise = TRUE", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>


<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-nomogram_1-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-nomogram_1-code-editor`)), session)
output$`tutorial-exercise-nomogram_1-output` <- renderUI({
  `tutorial-exercise-nomogram_1-result`()
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "nomogram_1", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "nomogram_1", code = "df$Suburb |> print()", 
        opts = list(label = "\"nomogram_1\"", exercise = "TRUE"), 
        engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "main_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "main_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = TRUE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "nomogram_1", exercise = TRUE, code = "df$Suburb |> print()", 
        out.width.px = 624, out.height.px = 384, params.src = "nomogram_1, exercise = TRUE", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_radio", label = "nomogram_3", question = structure("Which term (predictor) in the model contributes <em>most<\u002fem> to the outcome for Otara East?", html = TRUE, class = c("html", 
"character")), answers = list(structure(list(id = "lnr_ans_fa15f16", 
    option = "`log_population`", value = "`log_population`", 
    label = structure("<code>log_population<\u002fcode>", html = TRUE, class = c("html", 
    "character")), correct = FALSE, message = structure("No, if you identify the red dot in the graph for <code>log_population<\u002fcode> and read vertically up from it, you’ll touch the 2nd red dot (4th furthest) from 0 in the top line (<code>β(X-m) terms<\u002fcode>). So <code>log_population<\u002fcode> will contribute to the <em>4th most<\u002fem> to the outcome. Please try again", html = TRUE, class = c("html", 
    "character")), type = "literal"), class = c("tutorial_question_answer", 
"tutorial_quiz_answer")), structure(list(id = "lnr_ans_901c49f", 
    option = "`Maori_proportion`", value = "`Maori_proportion`", 
    label = structure("<code>Maori_proportion<\u002fcode>", html = TRUE, class = c("html", 
    "character")), correct = FALSE, message = structure("No, if you identify the red dot in the graph for <code>Maori_proportion<\u002fcode> and read vertically up from it, you’ll touch the red dot that’s closest from 0 in the top line (<code>β(X-m) terms<\u002fcode>). So <code>Maori_proportion<\u002fcode> will contribute the <em>least<\u002fem> to the outcome. Please try again", html = TRUE, class = c("html", 
    "character")), type = "literal"), class = c("tutorial_question_answer", 
"tutorial_quiz_answer")), structure(list(id = "lnr_ans_d95fd53", 
    option = "`Pacific_proportion`", value = "`Pacific_proportion`", 
    label = structure("<code>Pacific_proportion<\u002fcode>", html = TRUE, class = c("html", 
    "character")), correct = TRUE, message = structure("Yes, if you identify the red dot that’s furthest from 0 in the top line – β(X-m) terms – then read vertically down from it, you’ll touch the red dot in the graph for <code>Pacific_proportion<\u002fcode>", html = TRUE, class = c("html", 
    "character")), type = "literal"), class = c("tutorial_question_answer", 
"tutorial_quiz_answer")), structure(list(id = "lnr_ans_c9805d0", 
    option = "`nzDepQuint`", value = "`nzDepQuint`", label = structure("<code>nzDepQuint<\u002fcode>", html = TRUE, class = c("html", 
    "character")), correct = FALSE, message = structure("No, if you identify the red dot in the graph for <code>nzDepQuint<\u002fcode> and read vertically up from it, you’ll touch the 4th red dot (2nd furthest) from 0 in the top line (<code>β(X-m) terms<\u002fcode>). So <code>nzDepQuint<\u002fcode> will contribute the <em>2nd most<\u002fem> to the outcome. Please try again", html = TRUE, class = c("html", 
    "character")), type = "literal"), class = c("tutorial_question_answer", 
"tutorial_quiz_answer")), structure(list(id = "lnr_ans_7ebb25", 
    option = "`Permethrin_rate`", value = "`Permethrin_rate`", 
    label = structure("<code>Permethrin_rate<\u002fcode>", html = TRUE, class = c("html", 
    "character")), correct = FALSE, message = structure("No, if you identify the red dot in the graph for <code>Permethrin_rate<\u002fcode> and read vertically up from it, you’ll touch the 3<sup>rd<\u002fsup> red dot (3<sup>rd<\u002fsup> furthest) from 0 in the top line (<code>β(X-m) terms<\u002fcode>). So <code>Permethrin_rate<\u002fcode> will contribute the <em>3rd most<\u002fem> to the outcome. Please try again", html = TRUE, class = c("html", 
    "character")), type = "literal"), class = c("tutorial_question_answer", 
"tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
"character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
"character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
"character")), try_again = structure("Incorrect", html = TRUE, class = c("html", 
"character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
"character")), message = NULL, post_message = NULL), ids = list(
    answer = "nomogram_3-answer", question = "nomogram_3"), loading = NULL, 
    random_answer_order = FALSE, allow_retry = TRUE, seed = 85258104.4602986, 
    options = list()), class = c("learnr_radio", "tutorial_question"
)), session = session)
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_radio", label = "nomogram_4", question = structure("Which of the following is much higher for Otara East compared to the suburb averages?", html = TRUE, class = c("html", 
"character")), answers = list(structure(list(id = "lnr_ans_dedb010", 
    option = "`Total Score` and all predictors", value = "`Total Score` and all predictors", 
    label = structure("<code>Total Score<\u002fcode> and all predictors", html = TRUE, class = c("html", 
    "character")), correct = FALSE, message = structure("No, this is not true for all predictors: the central position of the red dot in the plot for <code>log_population<\u002fcode> indicates that the population for Otara East is similar to the median population (for all suburbs). Please try again", html = TRUE, class = c("html", 
    "character")), type = "literal"), class = c("tutorial_question_answer", 
"tutorial_quiz_answer")), structure(list(id = "lnr_ans_5511184", 
    option = "`Total Score` and all predictors except `log_population`", 
    value = "`Total Score` and all predictors except `log_population`", 
    label = structure("<code>Total Score<\u002fcode> and all predictors except <code>log_population<\u002fcode>", html = TRUE, class = c("html", 
    "character")), correct = TRUE, message = structure("Yes, for these terms, the red dot in each plot is to the far right of the distribution for all suburbs. The high <code>Total Score<\u002fcode> for Otara East (4.22), which equates to an ARF count of 5.52, is due to high average values for all predictors except <code>log_population<\u002fcode>", html = TRUE, class = c("html", 
    "character")), type = "literal"), class = c("tutorial_question_answer", 
"tutorial_quiz_answer")), structure(list(id = "lnr_ans_f3d6018", 
    option = "All predictors except `log_population` but not `Total Score`", 
    value = "All predictors except `log_population` but not `Total Score`", 
    label = structure("All predictors except <code>log_population<\u002fcode> but not <code>Total Score<\u002fcode>", html = TRUE, class = c("html", 
    "character")), correct = FALSE, message = structure("No, the red dot in the <code>Total score<\u002fcode> plot is to the far right of the distribution for all suburbs. Please try again", html = TRUE, class = c("html", 
    "character")), type = "literal"), class = c("tutorial_question_answer", 
"tutorial_quiz_answer")), structure(list(id = "lnr_ans_ce7428f", 
    option = "`Total Score` only", value = "`Total Score` only", 
    label = structure("<code>Total Score<\u002fcode> only", html = TRUE, class = c("html", 
    "character")), correct = FALSE, message = structure("No, the red dot is, not only to the far right of the plot for <code>Total Score<\u002fcode>, but also for most predictors. Please try again", html = TRUE, class = c("html", 
    "character")), type = "literal"), class = c("tutorial_question_answer", 
"tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
"character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
"character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
"character")), try_again = structure("Incorrect", html = TRUE, class = c("html", 
"character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
"character")), message = NULL, post_message = NULL), ids = list(
    answer = "nomogram_4-answer", question = "nomogram_4"), loading = NULL, 
    random_answer_order = FALSE, allow_retry = TRUE, seed = 413669238.80737, 
    options = list()), class = c("learnr_radio", "tutorial_question"
)), session = session)
</script>


<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-model_fit_1-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-model_fit_1-code-editor`)), session)
output$`tutorial-exercise-model_fit_1-output` <- renderUI({
  `tutorial-exercise-model_fit_1-result`()
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "model_fit_1", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "model_fit_1", code = "car::marginalModelPlots(fit)", 
        opts = list(label = "\"model_fit_1\"", exercise = "TRUE"), 
        engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "main_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "main_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = TRUE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "model_fit_1", exercise = TRUE, code = "car::marginalModelPlots(fit)", 
        out.width.px = 624, out.height.px = 384, params.src = "model_fit_1, exercise = TRUE", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_radio", label = "model_fit_2", 
    question = structure("Which of the following terms for the model would be a POOR fit for the data?", html = TRUE, class = c("html", 
    "character")), answers = list(structure(list(id = "lnr_ans_2dbea5", 
        option = "`Permethrin_rate`", value = "`Permethrin_rate`", 
        label = structure("<code>Permethrin_rate<\u002fcode>", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = structure("No, that’s harsh! There’s only poor alignment between the <em>data<\u002fem> (blue) and <em>model<\u002fem> (red) curves for <code>Permethrin_rate<\u002fcode>&gt; 2500 but this is minor as there’s few observations in that extreme region. Please try again", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_dac1419", 
        option = "`Permethrin_rate` and `Pacific_proportion`", 
        value = "`Permethrin_rate` and `Pacific_proportion`", 
        label = structure("<code>Permethrin_rate<\u002fcode> and <code>Pacific_proportion<\u002fcode>", html = TRUE, class = c("html", 
        "character")), correct = FALSE, message = structure("No, that’s harsh! For <code>Permethrin_rate<\u002fcode>, there’s only poor alignment between the <em>data<\u002fem> (blue) and <em>model<\u002fem> (red) curves for Permethrin_rate&gt;2500 but this is minor as there’s few observations in that extreme region. For <code>Pacific_proportion<\u002fcode>, the <em>data<\u002fem> (blue) and <em>model<\u002fem> (red) curves are misaligned in some areas of the plot but the degree of deviation is minor. Please try again", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer")), structure(list(id = "lnr_ans_aa02d2a", 
        option = "None of these", value = "None of these", label = structure("None of these", html = TRUE, class = c("html", 
        "character")), correct = TRUE, message = structure("Yes, the <em>data<\u002fem> (blue) and <em>model<\u002fem> (red) curves are well-aligned for <code>Maori_proportion<\u002fcode> and the overall model (<code>Linear_Predictor<\u002fcode>). And although there is some misalignment in the other 2 plots, this is minor", html = TRUE, class = c("html", 
        "character")), type = "literal"), class = c("tutorial_question_answer", 
    "tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
    "character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
    "character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
    "character")), try_again = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
    "character")), message = NULL, post_message = NULL), ids = list(
        answer = "model_fit_2-answer", question = "model_fit_2"), 
    loading = NULL, random_answer_order = FALSE, allow_retry = TRUE, 
    seed = 313151980.354177, options = list()), class = c("learnr_radio", 
"tutorial_question")), session = session)
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-publication_1-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-publication_1-code-editor`)), session)
output$`tutorial-exercise-publication_1-output` <- renderUI({
  `tutorial-exercise-publication_1-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "publication_1", global_setup = structure(c("library(learnr)", 
"library(tufte)", "library(rio)", "library(visdat)", "library(fs)", 
"library(ggplot2)", "library(DiagrammeR)", "library(r2symbols) # cheatsheet here: https://r2symbols.obi.obianom.com/symbols/", 
"library(learnr)", "library(gradethis)", "library(skimr)", "library(lattice)", 
"library(epiDisplay)", "library(finalfit)", "library(vembedr)", 
"library(png)", "", "gradethis::gradethis_setup()", "#setwd(\"./data\")", 
"# load(\"./Cloud/data/ARF_permethrin.rdata\")", "load(\"./data/ARF_permethrin.rdata\")", 
"# library(readxl)", "# df <- read_xlsx(\"C:\\\\Users\\\\jslu002\\\\Desktop\\\\for_shiny.xlsx\")", 
"df <- data.frame(for_shiny)", "df <- df[!duplicated(df),]", 
"", "names(df) <- c(\"arf_count\", \"population\", \"Pacific_proportion\",", 
"               \"Maori_proportion\", \"Permethrin_rate\", \"Suburb\",", 
"               \"nzdep_decile\", \"arf_rate\")", "", "## 5 groups instead of 10", 
"df$nzDepQuint <- cut(df$nzdep_decile, breaks = c(-Inf, 2, 4, 6, 8, Inf),", 
"                     labels = c(\"1 to 2 (least deprived)\",", 
"                                \"3 to 4\",", "                                \"5 to 6\",", 
"                                \"7 to 8\",", "                                \"9 to 10 (most deprived)\"))", 
"", "df$log_population <- log(df$population)", "", "fit <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion, offset=log_population,", "           family = poisson(link = \"log\"),", 
"           data = df)", "", "fit1 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "fit5 <- glm(arf_count ~ Permethrin_rate +", 
"             nzDepQuint +", "             Pacific_proportion +", 
"             Maori_proportion + offset(log(population)),", "           family = quasipoisson(link = \"log\"),", 
"           data = df)", "", "mycounter <- 0", "Qnum <- function(count = mycounter){", 
"  mycounter <<- mycounter + 1", "  paste0(\"Question \", mycounter)", 
"}", "", "knitr::opts_chunk$set(error = TRUE)"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "publication_1", code = "df$Permethrin_rate_per_1000 <- df$Permethrin_rate/1000  # Like converting grams into kilograms\ndf$log_population <- log(df$population)  # Offset\n\ndependent <- \"arf_count\"\nexplanatory <- c(\"Pacific_proportion\", \"Maori_proportion\", \n                 \"Permethrin_rate_per_1000\",\"nzdep_decile\")\n\n# `fit_uni` creates univariable (unadjusted) models\nfit_uni = df |> \n  glmuni(dependent, explanatory, family = quasipoisson,\n         offset = df$log_population) |> \n  fit2df(estimate_name = \"Rate ratio (univariable)\")\n\n# `fit_multi` creates a multivariable (adjusted) model\nfit_multi = df |> \n  glmmulti(dependent, explanatory, family = quasipoisson, \n           offset = df$log_population) |> \n  fit2df(estimate_name = \"Rate ratio (multivariable)\")\n\ndf |>\n  ## Crosstable\n  summary_factorlist(dependent, explanatory, cont = \"median\", fit_id=TRUE)  |> \n  \n  #### Add univariable table\n  ff_merge(fit_uni, estimate_name = \"Rate ratio\") |> \n  \n  #### Add multivariable table\n  ff_merge(fit_multi, estimate_name = \"Rate ratio\",\n           last_merge = TRUE) |> \ndependent_label(df, dependent) -> regression_table\n\nregress_tab <- knitr::kable(regression_table, row.names = FALSE, \n                            align = c(\"l\", \"l\", \"r\", \"r\", \"r\"))\n\n#### Puts output into viewer so that it can be cut and pasted into Excel, \n#### then Word.\nregress_tab |> kableExtra::kable_styling()", 
        opts = list(label = "\"publication_1\"", exercise = "TRUE"), 
        engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "main_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "main_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = TRUE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "publication_1", exercise = TRUE, code = c("df$Permethrin_rate_per_1000 <- df$Permethrin_rate/1000  # Like converting grams into kilograms", 
        "df$log_population <- log(df$population)  # Offset", 
        "", "dependent <- \"arf_count\"", "explanatory <- c(\"Pacific_proportion\", \"Maori_proportion\", ", 
        "                 \"Permethrin_rate_per_1000\",\"nzdep_decile\")", 
        "", "# `fit_uni` creates univariable (unadjusted) models", 
        "fit_uni = df |> ", "  glmuni(dependent, explanatory, family = quasipoisson,", 
        "         offset = df$log_population) |> ", "  fit2df(estimate_name = \"Rate ratio (univariable)\")", 
        "", "# `fit_multi` creates a multivariable (adjusted) model", 
        "fit_multi = df |> ", "  glmmulti(dependent, explanatory, family = quasipoisson, ", 
        "           offset = df$log_population) |> ", "  fit2df(estimate_name = \"Rate ratio (multivariable)\")", 
        "", "df |>", "  ## Crosstable", "  summary_factorlist(dependent, explanatory, cont = \"median\", fit_id=TRUE)  |> ", 
        "  ", "  #### Add univariable table", "  ff_merge(fit_uni, estimate_name = \"Rate ratio\") |> ", 
        "  ", "  #### Add multivariable table", "  ff_merge(fit_multi, estimate_name = \"Rate ratio\",", 
        "           last_merge = TRUE) |> ", "dependent_label(df, dependent) -> regression_table", 
        "", "regress_tab <- knitr::kable(regression_table, row.names = FALSE, ", 
        "                            align = c(\"l\", \"l\", \"r\", \"r\", \"r\"))", 
        "", "#### Puts output into viewer so that it can be cut and pasted into Excel, ", 
        "#### then Word.", "regress_tab |> kableExtra::kable_styling()"
        ), out.width.px = 624, out.height.px = 384, params.src = "publication_1, exercise = TRUE", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::question_prerendered_chunk(structure(list(type = "learnr_radio", label = "pub_model", question = structure("What is the explanation for no significant association with <code>Permethrin_rate_per_1000<\u002fcode> and <code>arf_count<\u002fcode> in the multivariable model, but an association in the crude model?", html = TRUE, class = c("html", 
"character")), answers = list(structure(list(id = "lnr_ans_70c0b18", 
    option = "Multicollinearity", value = "Multicollinearity", 
    label = structure("Multicollinearity", html = TRUE, class = c("html", 
    "character")), correct = TRUE, message = structure("Yes, if we leave out the collinear variable <code>Pacific_proportion<\u002fcode>, we find an adjusted rate ratio between <code>Permethrin_rate_per_1000<\u002fcode> and the outcome is significant.", html = TRUE, class = c("html", 
    "character")), type = "literal"), class = c("tutorial_question_answer", 
"tutorial_quiz_answer")), structure(list(id = "lnr_ans_c39ab17", 
    option = "There really is no relationship between the two variables.", 
    value = "There really is no relationship between the two variables.", 
    label = structure("There really is no relationship between the two variables.", html = TRUE, class = c("html", 
    "character")), correct = FALSE, message = structure("No, think about the issue of the strong association between <code>Pacific_proportion<\u002fcode> and <code>Permethrin_rate_per_1000<\u002fcode>.", html = TRUE, class = c("html", 
    "character")), type = "literal"), class = c("tutorial_question_answer", 
"tutorial_quiz_answer")), structure(list(id = "lnr_ans_5ad2378", 
    option = "Correcting for confounding factors has resulted in no association.", 
    value = "Correcting for confounding factors has resulted in no association.", 
    label = structure("Correcting for confounding factors has resulted in no association.", html = TRUE, class = c("html", 
    "character")), correct = FALSE, message = structure("This is a possible explanation, and if we believed Pacific ethnicity to be a shared common cause of permethrin and rheumatic fever, it could be possible. In actual fact, given the very strong correlation between permethrin and Pacific ethnicity makes multi-collinearity more likely.", html = TRUE, class = c("html", 
    "character")), type = "literal"), class = c("tutorial_question_answer", 
"tutorial_quiz_answer"))), button_labels = list(submit = structure("<span data-i18n=\"button.questionsubmit\">Submit Answer<\u002fspan>", html = TRUE, class = c("html", 
"character")), try_again = structure("<span data-i18n=\"button.questiontryagain\">Try Again<\u002fspan>", html = TRUE, class = c("html", 
"character"))), messages = list(correct = structure("Correct!", html = TRUE, class = c("html", 
"character")), try_again = structure("Incorrect", html = TRUE, class = c("html", 
"character")), incorrect = structure("Incorrect", html = TRUE, class = c("html", 
"character")), message = NULL, post_message = NULL), ids = list(
    answer = "pub_model-answer", question = "pub_model"), loading = NULL, 
    random_answer_order = FALSE, allow_retry = TRUE, seed = 1683350233.71613, 
    options = list()), class = c("learnr_radio", "tutorial_question"
)), session = session)
</script>
</p>
<!--html_preserve-->
<script type="application/shiny-prerendered" data-context="dependencies">
{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["header-attrs"]},{"type":"character","attributes":{},"value":["2.26"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/pandoc"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["header-attrs.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.26"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["3.6.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/3.6.0"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery-3.6.0.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquerylib"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.1.4"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["bootstrap"]},{"type":"character","attributes":{},"value":["3.3.5"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/bootstrap"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["viewport"]}},"value":[{"type":"character","attributes":{},"value":["width=device-width, initial-scale=1"]}]},{"type":"character","attributes":{},"value":["js/bootstrap.min.js","shim/html5shiv.min.js","shim/respond.min.js"]},{"type":"character","attributes":{},"value":["css/cerulean.min.css"]},{"type":"character","attributes":{},"value":["<style>h1 {font-size: 34px;}\n       h1.title {font-size: 38px;}\n       h2 {font-size: 30px;}\n       h3 {font-size: 24px;}\n       h4 {font-size: 18px;}\n       h5 {font-size: 16px;}\n       h6 {font-size: 12px;}\n       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}\n       pre:not([class]) { background-color: white }<\/style>"]},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.26"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["pagedtable"]},{"type":"character","attributes":{},"value":["1.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/pagedtable-1.1"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["js/pagedtable.js"]},{"type":"character","attributes":{},"value":["css/pagedtable.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.26"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["highlightjs"]},{"type":"character","attributes":{},"value":["9.12.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/highlightjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["highlight.js"]},{"type":"character","attributes":{},"value":["textmate.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.26"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tutorial"]},{"type":"character","attributes":{},"value":["0.11.5.9000"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/tutorial"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tutorial.js"]},{"type":"character","attributes":{},"value":["tutorial.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["i18n"]},{"type":"character","attributes":{},"value":["21.6.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/i18n"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["i18next.min.js","tutorial-i18n-init.js"]},{"type":"NULL"},{"type":"character","attributes":{},"value":["<script id=\"i18n-cstm-trns\" type=\"application/json\">{\"language\":\"en\",\"resources\":{\"en\":{\"translation\":{\"button\":{\"runcode\":\"Run Code\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Hint\",\"hint_plural\":\"Hints\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Next Hint\",\"hintprev\":\"Previous Hint\",\"solution\":\"Solution\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Copy to Clipboard\",\"startover\":\"Start Over\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Continue\",\"submitanswer\":\"Submit Answer\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Previous Topic\",\"nexttopic\":\"Next Topic\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Try Again\"},\"text\":{\"startover\":\"Start Over\",\"areyousure\":\"Are you sure you want to start over? (All exercise progress will be reset)\",\"youmustcomplete\":\"You must complete the\",\"exercise\":\"exercise\",\"exercise_plural\":\"exercises\",\"inthissection\":\"in this section before continuing.\",\"code\":\"Code\",\"enginecap\":\"{{engine}} $t(text.code)\",\"quiz\":\"Quiz\",\"blank\":\"blank\",\"blank_plural\":\"blanks\",\"exercisecontainsblank\":\"This exercise contains {{count}} $t(text.blank).\",\"pleasereplaceblank\":\"Please replace {{blank}} with valid code.\",\"unparsable\":\"It looks like this might not be valid R code. R cannot determine how to turn your text into a complete command. You may have forgotten to fill in a blank, to remove an underscore, to include a comma between arguments, or to close an opening <code>&quot;<\\/code>, <code>'<\\/code>, <code>(<\\/code> or <code>{<\\/code> with a matching <code>&quot;<\\/code>, <code>'<\\/code>, <code>)<\\/code> or <code>}<\\/code>.\\n\",\"unparsablequotes\":\"<p>It looks like your R code contains specially formatted quotation marks or &quot;curly&quot; quotes (<code>{{character}}<\\/code>) around character strings, making your code invalid. R requires character values to be contained in straight quotation marks (<code>&quot;<\\/code> or <code>'<\\/code>).<\\/p> {{code}} <p>Don't worry, this is a common source of errors when you copy code from another app that applies its own formatting to text. You can try replacing the code on that line with the following. There may be other places that need to be fixed, too.<\\/p> {{suggestion}}\\n\",\"unparsableunicode\":\"<p>It looks like your R code contains an unexpected special character (<code>{{character}}<\\/code>) that makes your code invalid.<\\/p> {{code}} <p>Sometimes your code may contain a special character that looks like a regular character, especially if you copy and paste the code from another app. Try deleting the special character from your code and retyping it manually.<\\/p>\\n\",\"unparsableunicodesuggestion\":\"<p>It looks like your R code contains an unexpected special character (<code>{{character}}<\\/code>) that makes your code invalid.<\\/p> {{code}} <p>Sometimes your code may contain a special character that looks like a regular character, especially if you copy and paste the code from another app. You can try replacing the code on that line with the following. There may be other places that need to be fixed, too.<\\/p> {{suggestion}}\\n\",\"and\":\"and\",\"or\":\"or\",\"listcomma\":\", \",\"oxfordcomma\":\",\"}}},\"fr\":{\"translation\":{\"button\":{\"runcode\":\"Lancer le Code\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Indication\",\"hint_plural\":\"Indications\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Indication Suivante\",\"hintprev\":\"Indication Précédente\",\"solution\":\"Solution\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Copier dans le Presse-papier\",\"startover\":\"Recommencer\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Continuer\",\"submitanswer\":\"Soumettre\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Chapitre Précédent\",\"nexttopic\":\"Chapitre Suivant\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Réessayer\"},\"text\":{\"startover\":\"Recommencer\",\"areyousure\":\"Êtes-vous certains de vouloir recommencer? (La progression sera remise à zéro)\",\"youmustcomplete\":\"Vous devez d'abord compléter\",\"exercise\":\"l'exercice\",\"exercise_plural\":\"des exercices\",\"inthissection\":\"de cette section avec de continuer.\",\"code\":\"Code\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Quiz\",\"and\":\"et\",\"or\":\"ou\",\"oxfordcomma\":\"\"}}},\"es\":{\"translation\":{\"button\":{\"runcode\":\"Ejecutar código\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Pista\",\"hint_plural\":\"Pistas\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Siguiente pista\",\"hintprev\":\"Pista anterior\",\"solution\":\"Solución\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Copiar al portapapeles\",\"startover\":\"Reiniciar\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Continuar\",\"submitanswer\":\"Enviar respuesta\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Tema anterior\",\"nexttopic\":\"Tema siguiente\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Volver a intentar\"},\"text\":{\"startover\":\"Reiniciar\",\"areyousure\":\"¿De verdad quieres empezar de nuevo? (todo el progreso del ejercicio se perderá)\",\"youmustcomplete\":\"Debes completar\",\"exercise\":\"el ejercicio\",\"exercise_plural\":\"los ejercicios\",\"inthissection\":\"en esta sección antes de continuar.\",\"code\":\"Código\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Cuestionario\",\"and\":\"y\",\"or\":\"o\",\"oxfordcomma\":\"\"}}},\"pt\":{\"translation\":{\"button\":{\"runcode\":\"Executar código\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Dica\",\"hint_plural\":\"Dicas\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Próxima dica\",\"hintprev\":\"Dica anterior\",\"solution\":\"Solução\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Copiar para a área de transferência\",\"startover\":\"Reiniciar\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Continuar\",\"submitanswer\":\"Enviar resposta\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Tópico anterior\",\"nexttopic\":\"Próximo tópico\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Tentar novamente\"},\"text\":{\"startover\":\"Reiniciar\",\"areyousure\":\"Tem certeza que deseja começar novamente? (todo o progresso feito será perdido)\",\"youmustcomplete\":\"Você deve completar\",\"exercise\":\"o exercício\",\"exercise_plural\":\"os exercícios\",\"inthissection\":\"nesta seção antes de continuar.\",\"code\":\"Código\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Quiz\",\"and\":\"e\",\"or\":\"ou\",\"oxfordcomma\":\"\"}}},\"tr\":{\"translation\":{\"button\":{\"runcode\":\"Çalıştırma Kodu\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Ipucu\",\"hint_plural\":\"İpuçları\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Sonraki İpucu\",\"hintprev\":\"Önceki İpucu\",\"solution\":\"Çözüm\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Pano'ya Kopyala\",\"startover\":\"Baştan Başlamak\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Devam et\",\"submitanswer\":\"Cevabı onayla\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Önceki Konu\",\"nexttopic\":\"Sonraki Konu\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Tekrar Deneyin\"},\"text\":{\"startover\":\"Baştan Başlamak\",\"areyousure\":\"Baştan başlamak istediğinizden emin misiniz? (tüm egzersiz ilerlemesi kaybolacak)\",\"youmustcomplete\":\"Tamamlamalısın\",\"exercise\":\"egzersiz\",\"exercise_plural\":\"egzersizler\",\"inthissection\":\"devam etmeden önce bu bölümde\",\"code\":\"Kod\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Sınav\",\"oxfordcomma\":\"\"}}},\"emo\":{\"translation\":{\"button\":{\"runcode\":\"🏃\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"💡\",\"hint_plural\":\"$t(button.hint)\",\"hinttitle\":\"$t(button.hint)\",\"solution\":\"🎯\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"📋\",\"startover\":\"⏮\",\"startovertitle\":\"Start Over\",\"continue\":\"✅\",\"submitanswer\":\"🆗\",\"submitanswertitle\":\"Submit Answer\",\"previoustopic\":\"⬅\",\"nexttopic\":\"➡\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"🔁\"},\"text\":{\"startover\":\"⏮\",\"areyousure\":\"🤔\",\"youmustcomplete\":\"⚠️ 👉 🧑‍💻\",\"exercise\":\"\",\"exercise_plural\":\"\",\"inthissection\":\"\",\"code\":\"💻\",\"enginecap\":\"$t(text.code) {{engine}}\",\"oxfordcomma\":\"\"}}},\"eu\":{\"translation\":{\"button\":{\"runcode\":\"Kodea egikaritu\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Laguntza\",\"hint_plural\":\"Laguntza\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Aurreko laguntza\",\"hintprev\":\"Hurrengo laguntza\",\"solution\":\"Ebazpena\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Arbelean kopiatu\",\"startover\":\"Berrabiarazi\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Jarraitu\",\"submitanswer\":\"Erantzuna bidali\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Aurreko atala\",\"nexttopic\":\"Hurrengo atala\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Berriro saiatu\"},\"text\":{\"startover\":\"Berrabiarazi\",\"areyousure\":\"Berriro hasi nahi duzu? (egindako lana galdu egingo da)\",\"youmustcomplete\":\"Aurrera egin baino lehen atal honetako\",\"exercise\":\"ariketa egin behar duzu.\",\"exercise_plural\":\"ariketak egin behar dituzu.\",\"inthissection\":\"\",\"code\":\"Kodea\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Galdetegia\",\"oxfordcomma\":\"\"}}},\"de\":{\"translation\":{\"button\":{\"runcode\":\"Code ausführen\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Tipp\",\"hint_plural\":\"Tipps\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Nächster Tipp\",\"hintprev\":\"Vorheriger Tipp\",\"solution\":\"Lösung\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"In die Zwischenablage kopieren\",\"startover\":\"Neustart\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Weiter\",\"submitanswer\":\"Antwort einreichen\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Vorheriges Kapitel\",\"nexttopic\":\"Nächstes Kapitel\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Nochmal versuchen\"},\"text\":{\"startover\":\"Neustart\",\"areyousure\":\"Bist du sicher, dass du neustarten willst? (der gesamte Lernfortschritt wird gelöscht)\",\"youmustcomplete\":\"Vervollstädinge\",\"exercise\":\"die Übung\",\"exercise_plural\":\"die Übungen\",\"inthissection\":\"in diesem Kapitel, bevor du fortfährst.\",\"code\":\"Code\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Quiz\",\"blank\":\"Lücke\",\"blank_plural\":\"Lücken\",\"pleasereplaceblank\":\"Bitte ersetze {{blank}} mit gültigem Code.\",\"unparsable\":\"Dies scheint kein gültiger R Code zu sein. R kann deinen Text nicht in einen gültigen Befehl übersetzen. Du hast vielleicht vergessen, die Lücke zu füllen, einen Unterstrich zu entfernen, ein Komma zwischen Argumente zu setzen oder ein eröffnendes <code>&quot;<\\/code>, <code>'<\\/code>, <code>(<\\/code> oder <code>{<\\/code> mit einem zugehörigen <code>&quot;<\\/code>, <code>'<\\/code>, <code>)<\\/code> oder <code>}<\\/code> zu schließen.\\n\",\"and\":\"und\",\"or\":\"oder\",\"listcomma\":\", \",\"oxfordcomma\":\",\"}}},\"ko\":{\"translation\":{\"button\":{\"runcode\":\"코드 실행\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"힌트\",\"hint_plural\":\"힌트들\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"다음 힌트\",\"hintprev\":\"이전 힌트\",\"solution\":\"솔루션\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"클립보드에 복사\",\"startover\":\"재학습\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"다음 학습으로\",\"submitanswer\":\"정답 제출\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"이전 토픽\",\"nexttopic\":\"다음 토픽\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"재시도\"},\"text\":{\"startover\":\"재학습\",\"areyousure\":\"다시 시작 하시겠습니까? (모든 예제의 진행 정보가 재설정됩니다)\",\"youmustcomplete\":\"당신은 완료해야 합니다\",\"exercise\":\"연습문제\",\"exercise_plural\":\"연습문제들\",\"inthissection\":\"이 섹션을 실행하기 전에\",\"code\":\"코드\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"퀴즈\",\"blank\":\"공백\",\"blank_plural\":\"공백들\",\"exercisecontainsblank\":\"이 연습문제에는 {{count}}개의 $t(text.blank)이 포함되어 있습니다.\",\"pleasereplaceblank\":\"{{blank}}를 유효한 코드로 바꾸십시오.\",\"unparsable\":\"이것은 유효한 R 코드가 아닐 수 있습니다. R은 텍스트를 완전한 명령으로 변환하는 방법을 결정할 수 없습니다. 당신은 공백이나 밑줄을 대체하여 채우기, 인수를 컴마로 구분하기, 또는 <code>&quot;<\\/code>, <code>'<\\/code>, <code>(<\\/code> , <code>{<\\/code>로 시작하는 구문을 닫는 <code>&quot;<\\/code>, <code>'<\\/code>, <code>)<\\/code>, <code>}<\\/code>을 잊었을 수도 있습니다.\\n\",\"and\":\"그리고\",\"or\":\"혹은\",\"listcomma\":\", \",\"oxfordcomma\":\"\"}}},\"zh\":{\"translation\":{\"button\":{\"runcode\":\"运行代码\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"提示\",\"hint_plural\":\"提示\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"下一个提示\",\"hintprev\":\"上一个提示\",\"solution\":\"答案\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"复制到剪切板\",\"startover\":\"重新开始\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"继续\",\"submitanswer\":\"提交答案\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"上一专题\",\"nexttopic\":\"下一专题\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"再试一次\"},\"text\":{\"startover\":\"重置\",\"areyousure\":\"你确定要重新开始吗? (所有当前进度将被重置)\",\"youmustcomplete\":\"你必须完成\",\"exercise\":\"练习\",\"exercise_plural\":\"练习\",\"inthissection\":\"在进行本节之前\",\"code\":\"代码\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"测试\",\"blank\":\"空\",\"blank_plural\":\"空\",\"exercisecontainsblank\":\"本练习包含{{count}}个$t(text.blank)\",\"pleasereplaceblank\":\"请在{{blank}}内填写恰当的代码\",\"unparsable\":\"这似乎不是有效的R代码。 R不知道如何将您的文本转换为完整的命令。 您是否忘了填空，忘了删除下划线，忘了在参数之间包含逗号，或者是忘了用<code>&quot;<\\/code>, <code>'<\\/code>, <code>)<\\/code>,<code>}<\\/code>来封闭<code>&quot;<\\/code>, <code>'<\\/code>, <code>(<\\/code>。 or <code>{<\\/code>。\\n\",\"unparsablequotes\":\"<p>您的R代码中似乎含有特殊格式的引号，或者弯引号(<code>{{character}}<\\/code>) 在字符串前后，在R中字符串应该被直引号(<code>&quot;<\\/code> 或者 <code>'<\\/code>)包裹。<\\/p> {{code}} <p>别担心，该错误经常在复制粘贴包含格式的代码时遇到， 您可以尝试将该行中的代码替换为以下代码，也许还有其他地方需要修改。<\\/p> {{suggestion}}\\n\",\"unparsableunicode\":\"<p>您的代码中似乎包含有异常字符(<code>{{character}}<\\/code>),导致代码无效。<\\/p> {{code}} <p>有时候你的代码可能含有看似正常字符的特殊字符，特别是当你复制粘贴其他来源代码的时候。 请试着删除这些特殊字符,重新输入<\\/p>\\n\",\"unparsableunicodesuggestion\":\"<p>您的代码中似乎包含有异常字符(<code>{{character}}<\\/code>),导致代码无效。<\\/p> {{code}} <p>有时候你的代码可能含有看似正常字符的特殊字符，特别是当你复制粘贴其他来源代码的时候。 请试着删除这些特殊字符,重新输入<\\/p>\\n\",\"and\":\"且\",\"or\":\"或\",\"listcomma\":\",\",\"oxfordcomma\":\",\"}}},\"pl\":{\"translation\":{\"button\":{\"runcode\":\"Uruchom kod\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Podpowiedź\",\"hint_plural\":\"Podpowiedzi\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Następna podpowiedź\",\"hintprev\":\"Poprzednia podpowiedź\",\"solution\":\"Rozwiązanie\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Kopiuj do schowka\",\"startover\":\"Zacznij od początku\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Kontynuuj\",\"submitanswer\":\"Wyślij\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Poprzednia sekcja\",\"nexttopic\":\"Następna sekcja\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Spróbuj ponownie\"},\"text\":{\"startover\":\"Zacznij od początku\",\"areyousure\":\"Czy na pewno chcesz zacząć od początku? (cały postęp w zadaniu zostanie utracony)\",\"youmustcomplete\":\"Musisz ukończyć\",\"exercise\":\"ćwiczenie\",\"exercise_plural\":\"ćwiczenia\",\"inthissection\":\"w tej sekcji przed kontynuowaniem\",\"code\":\"Kod\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Quiz\",\"blank\":\"luka\",\"blank_plural\":\"luk(i)\",\"exercisecontainsblank\":\"To ćwiczenie zawiera {{count}} $t(text.blank).\",\"pleasereplaceblank\":\"Proszę uzupełnić {{blank}} prawidłowym kodem.\",\"unparsable\":\"Wygląda na to, że może to nie być prawidłowy kod R. R nie jest w stanie przetworzyć Twojego tekstu na polecenie. Mogłeś(-aś) zapomnieć wypełnić luki, usunąć podkreślnik, umieścić przecinka między argumentami, lub zamknąć znak <code>&quot;<\\/code>, <code>'<\\/code>, <code>(<\\/code> lub <code>{<\\/code> odpowiadającym <code>&quot;<\\/code>, <code>'<\\/code>, <code>)<\\/code> lub <code>}<\\/code>.\\n\",\"unparsablequotes\":\"<p>Wygląda na to, że Twój kod zawiera szczególnie sformatowane cudzysłowy lub cudzysłowy typograficzne (<code>{{character}}<\\/code>) przy ciągach znaków, co sprawia, że kod jest niepoprawny. R wymaga cudzysłowów prostych (<code>&quot;<\\/code> albo <code>'<\\/code>).<\\/p> {{code}} <p>Nie martw się, to powszechne źródło błędów, gdy kopiuje się kod z innego programu, który sam formatuje teskt. Możesz spróbować zastąpić swój kod następującym kodem. Mogą być też inne miejsca, które wymagają poprawienia.<\\/p> {{suggestion}}\\n\",\"unparsableunicode\":\"<p>Wygląda na to, że Twój kod zawiera niespodziewany znak specjalny (<code>{{character}}<\\/code>), co sprawia, że kod jest niepoprawny.<\\/p> {{code}} <p>Czasami Twój kod może zawierać znak specjalny, który wygląda jak zwykły znak, zwłaszcza jeśli kopiujesz kod z innego programu. Spróbuj usunąć znak specjalny i wpisać do ponownie ręcznie.<\\/p>\\n\",\"unparsableunicodesuggestion\":\"<p>Wygląda na to, że Twój kod zawiera niespodziewany znak specjalny (<code>{{character}}<\\/code>), co sprawia, że kod jest niepoprawny.<\\/p> {{code}} <p>Czasami Twój kod może zawierać znak specjalny, który wygląda jak zwykły znak, zwłaszcza jeśli kopiujesz kod z innego programu. Możesz spróbować zastąpić swój kod następującym kodem. Mogą być też inne miejsca, które wymagają poprawienia.<\\/p> {{suggestion}}\\n\",\"and\":\"i\",\"or\":\"lub\",\"listcomma\":\", \",\"oxfordcomma\":\"\"}}},\"no\":{\"translation\":{\"button\":{\"runcode\":\"Kjør kode\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Hint\",\"hint_plural\":\"Hint\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Neste hint\",\"hintprev\":\"Forrige hint\",\"solution\":\"Løsning\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Kopier til utklippstavle\",\"startover\":\"Start på nytt\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Fortsett\",\"submitanswer\":\"Send inn svar\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Forrige tema\",\"nexttopic\":\"Neste tema\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Prøv igjen\"},\"text\":{\"startover\":\"Start på nytt\",\"areyousure\":\"Er du sikker på at du vil starte på nytt? (All fremgang vil bli slettet)\",\"youmustcomplete\":\"Du må fullføre\",\"exercise\":\"oppgave\",\"exercise_plural\":\"oppgaver\",\"inthissection\":\"i denne seksjonen før du fortsetter.\",\"code\":\"kode\",\"enginecap\":\"{{engine}}-$t(text.code)\",\"quiz\":\"Quiz\",\"blank\":\"tomrom\",\"blank_plural\":\"tomrom\",\"exercisecontainsblank\":\"Denne oppgaven inneholder {{count}} $t(text.blank).\",\"pleasereplaceblank\":\"Erstatt {{blank}} med gyldig kode.\",\"unparsable\":\"Dette ser ikke ut til å være gyldig R-kode. Slik R ser det, utgjør ikke teksten en fullstendig kommando. Du kan ha glemt å fylle ut en luke, å fjerne en understrek, å inkludere et komma mellom argumenter, eller å lukke et åpnende <code>&quot;<\\/code>, <code>'<\\/code>, <code>(<\\/code> eller <code>{<\\/code> med et tilsvarende <code>&quot;<\\/code>, <code>'<\\/code>, <code>)<\\/code> eller <code>}<\\/code>.\\n\",\"unparsablequotes\":\"<p>Det ser ut som om R-koden din inneholder spesielt formaterte anførselstegn eller &quot;krøllete&quot; anførselstegn (<code>{{character}}<\\/code>) rundt tekststrenger, noe som gjør koden din ugyldig. R krever at tekstverdier skal være inneholdt i rette anførselstegn (<code>&quot;<\\/code> eller <code>'<\\/code>).<\\/p> {{code}} <p>Dette er et vanlig problem når du kopierer kode fra et annet program som bruker sin egen formatering på tekst.  Du kan prøve å erstatte koden i den linjen med følgende. Kanskje er det også andre steder som må fikses.<\\/p> {{suggestion}}\\n\",\"unparsableunicode\":\"<p>Det ser ut som om R-koden din inneholder et uventet spesialtegn (<code>{{character}}<\\/code>) som gjør koden din ugyldig.<\\/p> {{code}} <p>Noen ganger kan koden din inneholde et spesialtegn som ser ut som et vanlig tegn, spesielt hvis du kopierer og limer inn koden fra et annet program. Prøv å slette spesialtegnet fra koden din og skriv det inn på nytt for hånd.<\\/p>\\n\",\"unparsableunicodesuggestion\":\"<p>Det ser ut som om R-koden din inneholder et uventet spesialtegn (<code>{{character}}<\\/code>) som gjør koden din ugyldig.<\\/p> {{code}} <p>Noen ganger kan koden din inneholde et spesialtegn som ser ut som et vanlig tegn, spesielt hvis du kopierer og limer inn koden fra et annet program. Du kan prøve å erstatte koden i den linjen med følgende. Kanskje er det også andre steder som må fikses.<\\/p> {{suggestion}}\\n\",\"and\":\"og\",\"or\":\"eller\",\"listcomma\":\", \",\"oxfordcomma\":\"\"}}}}}<\/script>"]},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tutorial-format"]},{"type":"character","attributes":{},"value":["0.11.5.9000"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmarkdown/templates/tutorial/resources"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tutorial-format.js"]},{"type":"character","attributes":{},"value":["tutorial-format.css","rstudio-theme.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["3.6.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/3.6.0"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery-3.6.0.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquerylib"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.1.4"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["navigation"]},{"type":"character","attributes":{},"value":["1.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/navigation-1.1"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tabsets.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.26"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["highlightjs"]},{"type":"character","attributes":{},"value":["9.12.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/highlightjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["highlight.js"]},{"type":"character","attributes":{},"value":["default.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.26"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["3.6.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/3.6.0"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery-3.6.0.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquerylib"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.1.4"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["font-awesome"]},{"type":"character","attributes":{},"value":["6.4.2"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["fontawesome"]}]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["css/all.min.css","css/v4-shims.min.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["fontawesome"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.5.2"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["bootbox"]},{"type":"character","attributes":{},"value":["5.5.2"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/bootbox"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["bootbox.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["idb-keyvalue"]},{"type":"character","attributes":{},"value":["3.2.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/idb-keyval"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["idb-keyval-iife-compat.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[false]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tutorial"]},{"type":"character","attributes":{},"value":["0.11.5.9000"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/tutorial"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tutorial.js"]},{"type":"character","attributes":{},"value":["tutorial.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["vembedr"]},{"type":"character","attributes":{},"value":["0.1.5"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["vembedr"]}]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["css/vembedr.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["vembedr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.1.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["vembedr"]},{"type":"character","attributes":{},"value":["0.1.5"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["vembedr"]}]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["css/vembedr.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["vembedr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.1.5"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.10.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.11.5.9000"]}]}]}
</script>
<!--/html_preserve-->
<!--html_preserve-->
<script type="application/shiny-prerendered" data-context="execution_dependencies">
{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["packages"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["packages","version"]},"class":{"type":"character","attributes":{},"value":["data.frame"]},"row.names":{"type":"integer","attributes":{},"value":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113]}},"value":[{"type":"character","attributes":{},"value":["assertthat","backports","base","base64enc","beanplot","boot","broom","bslib","cachem","checkmate","cli","codetools","colorspace","commonmark","compiler","curl","datasets","DiagrammeR","digest","dplyr","ellipsis","epiDisplay","evaluate","fansi","fastmap","finalfit","fontawesome","forcats","foreach","foreign","fs","generics","ggplot2","glmnet","glue","gradethis","graphics","grDevices","grid","gtable","htmltools","htmlwidgets","httpuv","httr","iterators","jomo","jquerylib","jsonlite","knitr","later","lattice","learnr","lifecycle","lme4","magrittr","markdown","MASS","Matrix","methods","mice","mime","minqa","mitml","munsell","nlme","nloptr","nnet","pan","pillar","pkgconfig","png","promises","purrr","r2symbols","R6","RColorBrewer","Rcpp","regplot","repr","rio","rlang","rmarkdown","rpart","rprojroot","rstudioapi","sass","scales","shape","shiny","skimr","sm","splines","stats","stringi","stringr","survival","tibble","tidyr","tidyselect","tools","tufte","utf8","utils","vctrs","vembedr","vioplot","visdat","visNetwork","withr","xfun","xtable","yaml","zoo"]},{"type":"character","attributes":{},"value":["0.2.1","1.4.1","4.4.0","0.1-3","1.3.1","1.3-30","1.0.5","0.6.2","1.0.8","2.3.1","3.6.2","0.2-19","2.1-0","1.9.1","4.4.0","5.2.1","4.4.0","1.0.11","0.6.35","1.1.4","0.3.2","3.5.0.2","0.23","1.0.6","1.1.1","1.0.7","0.5.2","1.0.0","1.5.2","0.8-86","1.6.3","0.1.3","3.5.0","4.1-8","1.7.0","0.2.14","4.4.0","4.4.0","4.4.0","0.3.4","0.5.7","1.6.4","1.6.14","1.4.7","1.0.14","2.7-6","0.1.4","1.8.8","1.45","1.3.2","0.22-6","0.11.5.9000","1.0.4","1.1-35.1","2.0.3","1.12","7.3-60.2","1.6-5","4.4.0","3.16.0","0.12","1.2.6","0.4-5","0.5.0","3.1-164","2.0.3","7.3-19","1.9","1.9.0","2.0.3","0.1-8","1.2.1","1.0.2","1.4","2.5.1","1.1-3","1.0.12","1.1","1.1.7","1.0.1","1.1.3","2.26","4.1.23","2.0.4","0.16.0","0.4.9","1.3.0","1.4.6.1","1.8.0","2.1.5","2.2-6.0","4.4.0","4.4.0","1.8.3","1.5.1","3.5-8","3.2.1","1.3.1","1.2.1","4.4.0","0.13","1.2.4","4.4.0","0.6.5","0.1.5","0.4.0","0.6.0","2.1.2","3.0.0","0.43","1.8-4","2.3.8","1.8-12"]}]}]}
</script>
<!--/html_preserve-->
</div>

</article> <!-- topics -->

<div class="topicsContainer">
<div class="topicsPositioner">
<div class="band">
<div class="bandContent topicsListContainer">

<!-- begin doc-metadata -->
<div id="doc-metadata">
<h1 class="title toc-ignore" style="display:none;">POPLHLTH 304 tutorial
6: count regression</h1>
<h4 class="author"><em>Simon Thornley and John Sluyter</em></h4>
<h4 class="date"><em>5 April, 2024</em></h4>
</div>
<!-- end doc-metadata -->

</div> <!-- bandContent.topicsListContainer -->
</div> <!-- band -->
</div> <!-- topicsPositioner -->
</div> <!-- topicsContainer -->


</main> <!-- bandContent page -->
</div> <!-- pageContent band -->



<!-- Build Tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("section-TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<script>
// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>


</body>

</html>
